--- modules/user/user.module	2014-04-03 11:03:31.334056000 +0200
+++ modules/user/user.module	2014-04-03 11:09:16.439517000 +0200
@@ -501,6 +501,10 @@
         file_usage_delete($account->original->picture, 'user', 'user', $account->uid);
         file_delete($account->original->picture);
       }
+      // Fixes issue where user picture was deleted after calls to user_save().
+      if (is_object($account) && !is_object($account->picture) && $account->picture > 0) {
+        $account->picture = file_load($account->picture);
+      }
       $account->picture = empty($account->picture->fid) ? 0 : $account->picture->fid;
 
       // Do not allow 'uid' to be changed.
