Index: sites/all/modules/contributed/apachesolr/apachesolr.module
===================================================================
--- sites/all/modules/contributed/apachesolr/apachesolr.module	2012-02-16 08:47:32.000000000 +0100
+++ sites/all/modules/contributed/apachesolr/apachesolr.module	2012-02-16 11:02:41.000000000 +0100
@@ -1732,9 +1732,11 @@
   foreach ($entity_info as $entity_type => $info) {
     if ($info['apachesolr']['indexable']) {
       $supported = apachesolr_get_index_bundles($env_id, $entity_type);
-      foreach (array_keys($info['bundles']) as $bundle) {
-        if (in_array($bundle, $supported)) {
-          $entity_info[$entity_type]['bundles'][$bundle]['apachesolr']['index'] = TRUE;
+      if(isset($info['bundles'])) {
+        foreach (array_keys($info['bundles']) as $bundle) {
+          if (in_array($bundle, $supported)) {
+            $entity_info[$entity_type]['bundles'][$bundle]['apachesolr']['index'] = TRUE;
+          }
         }
       }
     }
