Index: sites/all/modules/contributed/apachesolr_attachments/apachesolr_attachments.module
===================================================================
--- sites/all/modules/contributed/apachesolr_attachments/apachesolr_attachments.module	2012-02-16 08:47:32.000000000 +0100
+++ sites/all/modules/contributed/apachesolr_attachments/apachesolr_attachments.module	2012-02-16 11:02:41.000000000 +0100
@@ -455,6 +455,7 @@
 }
 
 function apachesolr_attachments_field_attach_update($parent_entity_type, $parent_entity) {
+  if(isset($parent_entity->type)) {
   foreach (field_info_instances($parent_entity_type, $parent_entity->type) as $instance) {
     $field_info = field_info_field($instance['field_name']);
     if ($field_info['type'] == 'file') {
@@ -474,6 +475,7 @@
       }
     }
   }
+  }
 }
 
 function apachesolr_attachments_field_attach_delete($parent_entity_type, $parent_entity) {
