Index: chr.module
===================================================================
--- sites/all/modules/contributed/chr/chr.module  (revision 3135)
+++ sites/all/modules/contributed/chr/chr.module  (working copy)
@@ -199,10 +199,12 @@
     case 'PUT':
       $curl_opt[CURLOPT_PUT] = TRUE;
       break;
+    case 'HEAD':
+      $curl_opt[CURLOPT_NOBODY] = TRUE;
+      break;
     default:
-      $result->error = 'invalid method ' . $options['method'];
-      $result->code = -1004;
-      return $result;
+      $curl_opt[CURLOPT_HEADER] = TRUE;
+      $curl_opt[CURLOPT_CUSTOMREQUEST] = $options['method'];
   }
