diff --git sites/all/modules/contributed/chr/chr.module sites/all/modules/contributed/chr/chr.module
index 8d7e796..05bc4c6 100644
--- sites/all/modules/contributed/chr/chr.module
+++ sites/all/modules/contributed/chr/chr.module
@@ -187,7 +187,7 @@ function curl_http_request($url, array $options = array()) {
     case 'POST':
       $curl_opt[CURLOPT_POST] = TRUE;
       if (!empty($options['data'])) {
-        $curl_opt[CURLOPT_POSTFIELDS] = drupal_http_build_query($options['data']);
+        $curl_opt[CURLOPT_POSTFIELDS] = $options['data'];
       }
       break;
     case 'PUT':
