--- sites/all/modules/contributed/chr/chr.module
+++ sites/all/modules/contributed/chr/chr.module
@@ -193,6 +193,12 @@
     case 'PUT':
       $curl_opt[CURLOPT_PUT] = TRUE;
       break;
+    // FPFIS fix jira REST remoting
+    // https://answers.atlassian.com/questions/72475/jira-update-custom-fields-using-rest-php-and-curl
+    case 'PUT_JIRA':
+      $curl_opt[CURLOPT_CUSTOMREQUEST] = "PUT";
+      $curl_opt[CURLOPT_POSTFIELDS] = $options['data'];
+      break;
     case 'HEAD':
       $curl_opt[CURLOPT_NOBODY] = TRUE;
       break;
