prevent sanitize function to remove html tag attibutes
===================================================================
--- includes/common.inc	2012-04-02 15:29:38.000000000 +0200
+++ includes.common.inc	2012-04-02 14:31:53.000000000 +0200
@@ -1520,10 +1520,10 @@
   $xhtml_slash = $count ? ' /' : '';
 
   // Clean up attributes.
-  $attr2 = implode(' ', _filter_xss_attributes($attrlist));
-  $attr2 = preg_replace('/[<>]/', '', $attr2);
-  $attr2 = strlen($attr2) ? ' ' . $attr2 : '';
-
+  //$attr2 = implode(' ', _filter_xss_attributes($attrlist));
+  //$attr2 = preg_replace('/[<>]/', '', $attr2);
+  //$attr2 = strlen($attr2) ? ' ' . $attr2 : '';
+  $attr2 = $attrlist;
   return "<$elem$attr2$xhtml_slash>";
 }
 
