--- sites/all/modules/contributed/content_lock/content_lock.info
+++ sites/all/modules/contributed/content_lock/content_lock.info
@@ -11,7 +11,7 @@ files[] = views/content_lock.views
 files[] = views/views_handler_field_is_locked.inc
 files[] = views/views_handler_filter_is_locked.inc
 files[] = views/views_handler_sort_is_locked.inc
-configure = admin/settings/content_lock
+configure = admin/config/content/content_lock
 
 ; Information added by drupal.org packaging script on 2012-07-12
 version = "7.x-1.4"
--- sites/all/modules/contributed/content_lock/modules/content_lock_timeout/content_lock_timeout.info
+++ sites/all/modules/contributed/content_lock/modules/content_lock_timeout/content_lock_timeout.info
@@ -5,7 +5,7 @@ project = "content_lock"
 dependencies[] = content_lock
 files[] = content_lock_timeout.install
 files[] = content_lock_timeout.module
-configure = admin/settings/content_lock
+configure = admin/config/content/content_lock
 
 ; Information added by drupal.org packaging script on 2012-07-12
 version = "7.x-1.4"
