--- sites/all/modules/contributed/file_entity/file_entity.file.inc	2013-10-21 15:34:18.000000000 +0200
+++ sites/all/modules/contributed/file_entity/file_entity.file.inc	2013-10-21 15:35:43.000000000 +0200
@@ -112,19 +112,18 @@
         ))
         ->execute();
     }
-  }
 
-  if (file_entity_file_get_mimetype_type($file) == 'image' && module_exists('image')) {
-    // If the image dimensions have changed, update any image field references
-    // to this file and flush image style derivatives.
-    if ($file->metadata['width'] != $file->metadata['width'] || $file->metadata['height'] != $file->metadata['height']) {
-      _file_entity_update_image_field_dimensions($file);
-    }
+    if (file_entity_file_get_mimetype_type($file) == 'image' && module_exists('image')) {
+      // If the image dimensions have changed, update any image field references
+      // to this file and flush image style derivatives.
+      if ($file->metadata['width'] != $file->metadata['width'] || $file->metadata['height'] != $file->metadata['height']) {
+        _file_entity_update_image_field_dimensions($file);
+      }
 
-    // Flush image style derivatives whenever an image is updated.
-    image_path_flush($file->uri);
+      // Flush image style derivatives whenever an image is updated.
+      image_path_flush($file->uri);
+    }
   }
-
   // Clear any related field caches.
   file_entity_invalidate_field_caches($file);
 }
