--- sites/all/modules/contributed/i18n/i18n_field/i18n_field.module	2014-03-27 13:38:06.990195000 +0100
+++ sites/all/modules/contributed/i18n/i18n_field/i18n_field.module	2014-03-27 13:42:21.217268000 +0100
@@ -280,6 +280,10 @@
       $on_value = $alter_element['#on_value'];
       $alter_element['#options'];
       $alter_element['#title'] = $alter_element['#options'][$on_value];
+      // For using label instead of "On value".
+      if ($instance['widget']['settings']['display_label']) {
+        $alter_element['#title'] = $instance_current['label'];
+      }
     }
   }
 
