--- sites/all/modules/contributed/i18n/i18n_taxonomy/i18n_taxonomy.module	2012-09-14 15:47:08.835066208 +0200
+++ sites/all/modules/contributed/i18n/i18n_taxonomy/i18n_taxonomy.module	2012-09-14 15:48:38.000000000 +0200
@@ -627,7 +627,7 @@
  */
 function i18n_taxonomy_taxonomy_vocabulary_update($vocabulary) {
   // Update language for related terms
-  switch ($vocabulary->i18n_mode) {
+  switch (i18n_taxonomy_vocabulary_mode($vocabulary)) {
     case I18N_MODE_LANGUAGE:
       $update['language'] = $vocabulary->language;
       break;
