--- sites/all/modules/contributed/linkchecker.module	2013-02-14 13:40:46.000000000 +0100
+++ sites/all/modules/contributed/linkchecker.module	2013-07-15 16:44:30.000000000 +0200
@@ -1538,7 +1538,8 @@
   // Collect the fields from this entity_type and bundle.
   foreach (field_info_instances($entity_type, $bundle_name) as $field_name => $instance) {
     $field = field_info_field($field_name);
-    $entity_field = $entity->$field['field_name'];
+    // #1923328: field_name array may be missing in $entity if field is empty.
+    $entity_field = !empty($entity->$field['field_name']) ? $entity->$field['field_name'] : array();;
 
     switch ($field['type']) {
       // Core fields.
@@ -1567,7 +1568,7 @@
       case 'link_field':
         foreach ($entity_field as $language) {
           foreach ($language as $item) {
-            $options = drupal_parse_url($item['url']);
+            $options = drupal_parse_url(link_cleanup_url($item['url']));
             $title = !empty($item['title']) ? $item['title'] : '';
             $text_items[] = $text_items_by_field[$field['field_name']][] = l($title, $options['path'], $options);
             $text_items[] = $text_items_by_field[$field['field_name']][] = _linkchecker_check_markup($title, NULL, linkchecker_entity_language($entity_type, $entity), TRUE);
