--- sites/all/modules/contributed/link/link.module	2013-02-26 14:02:16.000000000 +0100
+++ sites/all/modules/contributed/link/link.module	2013-02-26 14:11:23.000000000 +0100
@@ -345,6 +345,9 @@
   // Trim whitespace from URL.
   $item['url'] = trim($item['url']);
 
+  // add protocol if it's missing
+  $item['url'] = link_cleanup_url($item['url']);
+
   // If no attributes are set then make sure $item['attributes'] is an empty
   // array, so $field['attributes'] can override it.
   if (empty($item['attributes'])) {
