--- sites/all/modules/contributed/media/js/plugins/media.views.js	2012-11-11 03:10:00.000000000 +0100
+++ sites/all/modules/contributed/media/js/plugins/media.views.js	2013-06-13 14:55:28.000000000 +0200
@@ -67,7 +67,7 @@ Drupal.media.browser.views.select = func
 Drupal.media.browser.views.setup = function(view) {
   // Catch the click on a media item
   $('.view-content .media-item', view).bind('click', function () {
-    var fid = $(this).closest('a[data-fid]').data('fid'),
+    var fid = $(this).closest('.media-item[data-fid]').data('fid'),
       selectedFiles = new Array();
 
     // Remove all currently selected files
@@ -98,7 +98,7 @@ Drupal.media.browser.views.setup = funct
           selectedFiles.push(Drupal.media.browser.selectedMedia[index]);
 
           // Mark it as selected
-          $('.view-content *[data-fid=' + currentFid + '] .media-item', view).addClass('selected');
+          $('.view-content *[data-fid=' + currentFid + '].media-item', view).addClass('selected');
         }
       }
     }
