--- sites/all/modules/contributed/media_youtube/includes/MediaYouTubeStreamWrapper.inc	2013-07-31 12:10:35.232862910 +0200
+++ sites/all/modules/contributed/media_youtube/includes/MediaYouTubeStreamWrapper.inc	2013-07-31 12:11:35.079332307 +0200
@@ -50,7 +50,7 @@
       $dirname = drupal_dirname($local_path);
       file_prepare_directory($dirname, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
       $response = drupal_http_request($this->getOriginalThumbnailPath());
-      if (!isset($response->error)) {
+      if(strlen(trim($response->error)) == 0) {
         file_unmanaged_save_data($response->data, $local_path, TRUE);
       }
       else {
