--- sites/all/modules/contributed/notifications/notifications_account/notifications_account.pages.inc	2012-05-22 13:49:39.000000000 +0200
+++ sites/all/modules/contributed/notifications/notifications_account/notifications_account.pages.inc	2012-05-22 14:53:05.000000000 +0200
@@ -45,6 +45,7 @@
   $tips['edit'] = l(t('Edit your notifications settings'), "$path/edit");
 
   // Enable / disable all subscriptions
+  /*
   if (notifications_access_user($account, 'maintain')) {
     if (!empty($status[Notifications_Subscription::STATUS_ACTIVE])) {
       $tips['disable'] = l(t('Temporarily disable all your subscriptions'), "$path/notifications/update/disable");
@@ -53,6 +54,8 @@
       $tips['enable'] = l(t('Enable all your subscriptions'), "$path/notifications/update/enable");
     }
   }
+  */
+
   /*
   $link = notifications_get_link('unsubscribe', array('uid' => $account->uid, 'destination' => TRUE));
   $tips['cancel'] = l(t('Cancel all your subscriptions'), $link['href'], $link['options']);
--- sites/all/modules/contributed/notifications/notifications.pages.inc	2012-05-22 13:49:21.000000000 +0200
+++ sites/all/modules/contributed/notifications/notifications.pages.inc	2012-05-22 14:52:54.000000000 +0200
@@ -106,7 +106,7 @@
       $options = array();
       if (notifications_access_user($account, 'maintain')) {
         $options[] = l(t('Administer your subscriptions'), "user/$user->uid/notifications");
-        $options[] = l(t('Temporarily disable all your subscriptions'), "user/$user->uid/notifications/update/disable");
+        //$options[] = l(t('Temporarily disable all your subscriptions'), "user/$user->uid/notifications/update/disable");
       }
       $options[] = notifications_build_link('unsubscribe', array('title' => t('Cancel all your subscriptions')), 'user', $user, 'link');
       $output .= theme('item_list', $options);
