--- sites/all/modules/contributed/password_policy/password_policy.js	2013-02-27 10:53:26.000000000 +0100
+++ sites/all/modules/contributed/password_policy/password_policy.js	2013-02-27 10:40:29.000000000 +0100
@@ -10,7 +10,7 @@ Drupal.behaviors.passwordOverride = {
       strength: 0,
       message: '',
       indicatorText: ''
-    },
+    }
 
     // We take over the keyup function on password and instead make a call to 
     // the server to evaluate the password. When we get the status back we
@@ -37,7 +37,7 @@ Drupal.behaviors.passwordOverride = {
     Drupal.evaluatePasswordStrength = function (password, translate) {
       return pw_status;
     };
-  },
+  }
 };
 
 // We are overriding the normal evaluatePasswordStrength and instead are
