--- sites/all/modules/contributed/password_policy/password_policy.module	2013-03-29 14:51:41.000000000 +0100
+++ sites/all/modules/contributed/password_policy/password_policy.module	2013-03-29 14:52:21.000000000 +0100
@@ -222,14 +222,16 @@
 }
 
 function password_policy_form_alter(&$form, $form_state, $form_id) {
-
+  global $user;
   switch ($form_id) {
     case 'user_register_form':
     case 'user_profile_form':
+    if ($user->uid) {
       password_policy_password_element_alter($form['account']['pass'], $form['#user']);
       $form['#validate'][] = 'password_policy_form_validate';
       $form['#submit'][] = 'password_policy_password_submit';
       break;
+    }
   }
 }
 
