--- sites/all/modules/contributed/password_policy/password_policy.module	2013-03-29 15:01:34.000000000 +0100
+++ sites/all/modules/contributed/password_policy/password_policy.module	2013-03-29 15:15:18.000000000 +0100
@@ -221,15 +221,29 @@
   );
 }
 
-function password_policy_form_user_profile_form_alter(&$form, $form_state) {
-  password_policy_password_element_alter($form['account']['pass'], $form['#user']);
-  $form['#validate'][] = 'password_policy_user_profile_form_validate';
-  $form['#submit'][] = 'password_policy_password_submit';
+function password_policy_form_alter(&$form, $form_state, $form_id) {
+  
+  global $user;
+  switch ($form_id) {
+    case 'user_register_form':
+    case 'user_profile_form':
+      if (isset($form['#description'])) {
+        password_policy_password_element_alter($form['account']['pass'], $form['#user']);
+        $form['#validate'][] = 'password_policy_form_validate';
+        $form['#submit'][] = 'password_policy_password_submit';
+        break;
+     }
+  }
 }
 
-function password_policy_user_profile_form_validate($form, &$form_state) {
+function password_policy_form_validate($form, &$form_state) {
   if (!empty($form_state['values']['pass'])) {
     $account = $form['#user'];
+    unset($account->roles);
+    $newroles = $form_state['values']['roles'];
+    foreach ($newroles as $key => $value){
+      if ($value == TRUE) $account->roles[$key] = TRUE ;
+    }
     $policies = PasswordPolicy::matched_policies($account);
     $errors = array();
     foreach ($policies as $policy) {
