--- sites/default/default.settings.php	2012-09-19 11:06:04.736421020 +0200
+++ sites/default/default.settings.php	2012-09-19 11:14:44.068341179 +0200
@@ -496,6 +496,24 @@
 # drupal_fast_404();
 
 /**
+ * External access proxy settings:
+ *
+ * If your site must access the Internet via a web proxy then you can enter
+ * the proxy settings here. Currently only basic authentication is supported
+ * by using the username and password variables. The proxy_user_agent variable
+ * can be set to NULL for proxies that require no User-Agent header or to a
+ * non-empty string for proxies that limit requests to a specific agent. The
+ * proxy_exceptions variable is an array of host names to be accessed directly,
+ * not via proxy.
+ */
+
+ $proxy_settings = dirname(__FILE__) . '/proxy.settings.php';
+ if (file_exists($proxy_settings)) {
+   include $proxy_settings;
+ }
+
+
+/**
  * Authorized file system operations:
  *
  * The Update manager module included with Drupal provides a mechanism for
