--- modules/search/search.module	2013-03-26 09:05:14.000000000 +0100
+++ modules/search/search.module	2013-03-26 09:15:44.000000000 +0100
@@ -1067,7 +1067,7 @@
   $hidden = array();
   // Provide variables named after form keys so themers can print each element independently.
   foreach (element_children($variables['form']) as $key) {
-    $type = $variables['form'][$key]['#type'];
+    $type = (isset($variables['form'][$key]['#type']) ? $variables['form'][$key]['#type'] : '');
     if ($type == 'hidden' || $type == 'token') {
       $hidden[] = drupal_render($variables['form'][$key]);
     }
