--- sites/all/modules/contributed/tmgmt/ui/tmgmt_ui.module
+++ sites/all/modules/contributed/tmgmt/ui/tmgmt_ui.module
@@ -566,7 +566,7 @@ function _tmgmt_ui_review_form_element($data, TMGMTJobItem $job_item, &$zebra) {
       }
       $form[$target_key]['translation'] = array(
         '#type' => 'textarea',
-        '#default_value' => $data[$key]['#translation']['#text'],
+        '#default_value' => isset($data[$key]['#translation']['#text']) ? $data[$key]['#translation']['#text'] : NULL,
         '#title' => t('Translation'),
         '#disabled' => $job_item->isAccepted(),
       );
