--- sites/all/modules/contributed/uuid/uuid.tokens.inc	2012-10-08 12:28:13.281449660 +0200
+++ sites/all/modules/contributed/uuid/uuid.tokens.inc	2012-10-08 12:28:31.634997817 +0200
@@ -45,6 +45,7 @@
  * Implements hook_tokens().
  */
 function uuid_tokens($type, $tokens, array $data = array(), array $options = array()) {
+  $replacements = array();
   switch ($type) {
     case 'node':
       $node = $data['node'];
@@ -93,4 +94,4 @@
   }
 
   return $replacements;
-}
\ No newline at end of file
+}
