Modify the Workbench views to display revision title
Index: sites/all/modules/contributed/workbench_moderation/views/workbench_moderation.view.inc
===================================================================
--- sites/all/modules/contributed/workbench_moderation/views/workbench_moderation.view.inc	2012-01-14 00:12:30.000000000 +0100
+++ sites/all/modules/contributed/workbench_moderation/views/workbench_moderation.view.inc	2012-02-21 10:44:29.000000000 +0100


@@ -37,87 +37,77 @@
 $handler->display->display_options['pager']['options']['expose']['items_per_page_options'] = '10, 25, 50, 100, 200';
 $handler->display->display_options['style_plugin'] = 'table';
 $handler->display->display_options['style_options']['columns'] = array(
+  'nid' => 'nid',
   'nid_1' => 'nid_1',
   'log' => 'log',
-  'workbench_moderation_history_link' => 'workbench_moderation_history_link',
+  'nothing' => 'nothing',
   'state' => 'state',
   'moderation_actions' => 'moderation_actions',
   'title' => 'title',
-  'title_1' => 'title_1',
   'type' => 'type',
   'name' => 'name',
   'changed' => 'changed',
 );
 $handler->display->display_options['style_options']['default'] = 'changed';
 $handler->display->display_options['style_options']['info'] = array(
+  'nid' => array(
+    'sortable' => 0,
+    'default_sort_order' => 'asc',
+    'align' => '',
+    'separator' => '',
+  ),
   'nid_1' => array(
     'sortable' => 0,
     'default_sort_order' => 'asc',
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
   'log' => array(
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
-  'workbench_moderation_history_link' => array(
+  'nothing' => array(
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
   'state' => array(
     'sortable' => 1,
     'default_sort_order' => 'asc',
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
   'moderation_actions' => array(
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
   'title' => array(
     'sortable' => 1,
     'default_sort_order' => 'asc',
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
-  ),
-  'title_1' => array(
-    'sortable' => 1,
-    'default_sort_order' => 'asc',
-    'align' => '',
-    'separator' => '',
-    'empty_column' => 0,
   ),
   'type' => array(
     'sortable' => 1,
     'default_sort_order' => 'asc',
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
   'name' => array(
     'sortable' => 1,
     'default_sort_order' => 'asc',
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
   'changed' => array(
     'sortable' => 1,
     'default_sort_order' => 'desc',
     'align' => '',
     'separator' => '',
-    'empty_column' => 0,
   ),
 );
 $handler->display->display_options['style_options']['override'] = 1;
 $handler->display->display_options['style_options']['sticky'] = 0;
-$handler->display->display_options['style_options']['empty_table'] = 0;
+$handler->display->display_options['style_options']['order'] = 'desc';
 /* No results behavior: Global: Text area */
 $handler->display->display_options['empty']['area']['id'] = 'area';
 $handler->display->display_options['empty']['area']['table'] = 'views';
@@ -229,30 +219,21 @@
 $handler->display->display_options['fields']['moderation_actions']['element_default_classes'] = 1;
 $handler->display->display_options['fields']['moderation_actions']['hide_empty'] = 0;
 $handler->display->display_options['fields']['moderation_actions']['empty_zero'] = 0;
-/* Field: Content revision: Title */
-$handler->display->display_options['fields']['title_1']['id'] = 'title_1';
-$handler->display->display_options['fields']['title_1']['table'] = 'node_revision';
-$handler->display->display_options['fields']['title_1']['field'] = 'title';
-$handler->display->display_options['fields']['title_1']['alter']['alter_text'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['make_link'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['absolute'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['external'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['replace_spaces'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['trim_whitespace'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['nl2br'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['word_boundary'] = 1;
-$handler->display->display_options['fields']['title_1']['alter']['ellipsis'] = 1;
-$handler->display->display_options['fields']['title_1']['alter']['more_link'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['strip_tags'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['trim'] = 0;
-$handler->display->display_options['fields']['title_1']['alter']['html'] = 0;
-$handler->display->display_options['fields']['title_1']['element_label_colon'] = 1;
-$handler->display->display_options['fields']['title_1']['element_default_classes'] = 1;
-$handler->display->display_options['fields']['title_1']['hide_empty'] = 0;
-$handler->display->display_options['fields']['title_1']['empty_zero'] = 0;
-$handler->display->display_options['fields']['title_1']['hide_alter_empty'] = 1;
-$handler->display->display_options['fields']['title_1']['link_to_node'] = 0;
-$handler->display->display_options['fields']['title_1']['link_to_node_revision'] = 1;
+/* Field: Content: Title */
+$handler->display->display_options['fields']['title']['id'] = 'title';
+$handler->display->display_options['fields']['title']['table'] = 'node';
+$handler->display->display_options['fields']['title']['field'] = 'title';
+$handler->display->display_options['fields']['title']['alter']['alter_text'] = 0;
+$handler->display->display_options['fields']['title']['alter']['make_link'] = 0;
+$handler->display->display_options['fields']['title']['alter']['absolute'] = 0;
+$handler->display->display_options['fields']['title']['alter']['word_boundary'] = 0;
+$handler->display->display_options['fields']['title']['alter']['ellipsis'] = 0;
+$handler->display->display_options['fields']['title']['alter']['strip_tags'] = 0;
+$handler->display->display_options['fields']['title']['alter']['trim'] = 0;
+$handler->display->display_options['fields']['title']['alter']['html'] = 0;
+$handler->display->display_options['fields']['title']['hide_empty'] = 0;
+$handler->display->display_options['fields']['title']['empty_zero'] = 0;
+$handler->display->display_options['fields']['title']['link_to_node'] = 1;
 /* Field: Content: Type */
 $handler->display->display_options['fields']['type']['id'] = 'type';
 $handler->display->display_options['fields']['type']['table'] = 'node';
