Modify the My drafts view to display only drafts
Index: sites/all/modules/contributed/workbench_moderation/views/workbench_moderation.view.inc
===================================================================
--- sites/all/modules/contributed/workbench_moderation/views/workbench_moderation.view.inc	Fri Aug 24 10:20:51 2012
+++ sites/all/modules/contributed/workbench_moderation/views/workbench_moderation.view.inc	Fri Aug 24 10:22:58 2012
@@ -461,9 +461,7 @@
 $handler->display->display_options['filters']['state']['field'] = 'state';
 $handler->display->display_options['filters']['state']['value'] = array(
   'draft' => 'draft',
-  'needs_review' => 'needs_review',
 );
-$handler->display->display_options['filters']['state']['exposed'] = TRUE;
 $handler->display->display_options['filters']['state']['expose']['operator_id'] = 'state_op';
 $handler->display->display_options['filters']['state']['expose']['label'] = 'State';
 $handler->display->display_options['filters']['state']['expose']['operator'] = 'state_op';
