diff --git a/README.txt b/README.txt
index 239cde9..a406cae 100644
--- a/README.txt
+++ b/README.txt
@@ -97,8 +97,7 @@ track the moderation history of a particular revision, right up through the
 point where it is published.
 
 Revisions are a linear; revision history may not fork. This means that only the
-latest revision--Workbench Moderation calls this the "current draft"--may be
-edited or moderated.
+latest revision may be edited or moderated.
 
 ----
 2.  Installation
diff --git a/workbench_moderation.module b/workbench_moderation.module
index b652b6f..6f4b901 100644
--- a/workbench_moderation.module
+++ b/workbench_moderation.module
@@ -1781,7 +1781,7 @@ function workbench_moderation_messages($context, $node = NULL) {
       'message' => check_plain(workbench_moderation_state_label($state['my_revision']->state)),
     );
     $info_block_messages[] = array(
-      'label' => t('Current draft'),
+      'label' => t('Most recent revision'),
       'message' => !empty($revision_current) ? t('Yes') : t('No'),
     );
 
