--- sites/all/modules/contributed/wysiwyg/editors/js/ckeditor-3.0.js.original
+++ sites/all/modules/contributed/wysiwyg/editors/js/ckeditor-3.0.js
@@ -200,7 +200,13 @@
                 }
                 if (selection.getType() == CKEDITOR.SELECTION_TEXT) {
                   if (CKEDITOR.env.ie) {
-                    data.content = selection.getNative().createRange().text;
+                   var native = selection.getNative();
+                    if (native.getSelection)
+                      data.content = native.getSelection();
+                    if (native.createRange)
+                      data.content = native.createRange().text;
+                    else
+                      data.content = native.toString();
 
                   }
                   else {
