--- modules/taxonomy/taxonomy-term.tpl.php	2013-04-11 08:31:32.712662979 +0200
+++ modules/taxonomy/taxonomy-term.tpl.php	2013-04-11 10:46:54.000000000 +0200
@@ -44,7 +44,7 @@
 <div id="taxonomy-term-<?php print $term->tid; ?>" class="<?php print $classes; ?>">
 
   <?php if (!$page): ?>
-    <h2><a href="<?php print $term_url; ?>"><?php print $term_name; ?></a></h2>
+    <h2><a href="<?php print filter_xss($term_url); ?>"><?php print filter_xss($term_name); ?></a></h2>
   <?php endif; ?>
 
   <div class="content">
