--- modules/forum/forum-topic-list.tpl.php.orig	2013-04-15 13:21:05.749929252 +0200
+++ modules/forum/forum-topic-list.tpl.php	2013-04-15 13:49:49.000000000 +0200
@@ -60,7 +60,7 @@
         <?php print $topic->comment_count; ?>
         <?php if ($topic->new_replies): ?>
           <br />
-          <a href="<?php print $topic->new_url; ?>"><?php print $topic->new_text; ?></a>
+          <a href="<?php print filter_xss($topic->new_url); ?>"><?php print filter_xss($topic->new_text); ?></a>
         <?php endif; ?>
       </td>
       <td class="last-reply"><?php print $topic->last_reply; ?></td>
