--- sites/all/modules/contributed/entity/entity.tpl.php	2013-04-11 14:56:59.221929195 +0200
+++ sites/all/modules/contributed/entity/entity.tpl.php	2013-04-12 08:24:58.000000000 +0200
@@ -33,7 +33,7 @@
   <?php if (!$page): ?>
     <h2<?php print $title_attributes; ?>>
       <?php if ($url): ?>
-        <a href="<?php print $url; ?>"><?php print $title; ?></a>
+        <a href="<?php print filter_xss($url); ?>"><?php print $title; ?></a>
       <?php else: ?>
         <?php print $title; ?>
       <?php endif; ?>
