diff --git a/editors/ckeditor.inc b/editors/ckeditor.inc
index fcf168e..802e789 100644
--- editors/ckeditor.inc
+++ editors/ckeditor.inc
@@ -78,11 +78,11 @@ function wysiwyg_ckeditor_version($editor) {
     // version:'CKEditor 3.0 SVN',revision:'3665'
     // version:'3.0 RC',revision:'3753'
     // version:'3.0.1',revision:'4391'
-    if (preg_match('@version:\'(?:CKEditor )?([\d\.]+)(?:.+revision:\'([\d]+))?@', $line, $version)) {
+    if (preg_match('@version:\"(?:CKEditor )?([\d\.]+)(?:.+revision:\"([\d]+))?@', $line, $version)) {
       fclose($library);
       // Version numbers need to have three parts since 3.0.1.
       $version[1] = preg_replace('/^(\d+)\.(\d+)$/', '${1}.${2}.0', $version[1]);
-      return $version[1] . '.' . $version[2];
+      return $version[1];
     }
     $max_lines--;
   }
