diff --git a/date_views/includes/date_views_filter_handler_simple.inc b/date_views/includes/date_views_filter_handler_simple.inc
index d41f30b..22fd5f9 100644
--- a/date_views/includes/date_views_filter_handler_simple.inc
+++ b/date_views/includes/date_views_filter_handler_simple.inc
@@ -81,7 +81,8 @@ class date_views_filter_handler_simple extends views_handler_filter_date {
     }
     // This a fixed date.
     else {
-      $default_date = $options['value'][$prefix];
+      //Fix the notice if default date is empty
+      $default_date = (empty($options['value'][$prefix])) ? '' : $options['value'][$prefix];
     }
     return $default_date;
   }
@@ -101,9 +102,14 @@ class date_views_filter_handler_simple extends views_handler_filter_date {
       if (empty($this->options['exposed'])) {
         return str_replace(' ', 'T', $this->date_default_value($prefix));
       }
-      elseif (isset($this->options['expose']['identifier']) && !isset($_GET[$this->options['expose']['identifier']])) {
+      elseif (isset($this->options['expose']['identifier']) && !isset($_REQUEST[$this->options['expose']['identifier']])) {
         return str_replace(' ', 'T', $this->date_default_value($prefix));
       }
+      elseif ($this->options['is_grouped'] && isset($_REQUEST[$this->options['expose']['identifier']])) {
+        $identifier = $_REQUEST[$this->options['expose']['identifier']];
+        $options = $this->options['group_info']['group_items'][$identifier]['value'][$prefix . '_group'];
+        return str_replace(' ', 'T', $this->date_default_value($prefix, $options));
+      }
     }
 
     return str_replace(' ', 'T', $input);
