--- modules/user/user.pages.inc	2014-09-29 09:23:40.000000000 +0200
+++ modules/user/user.pages.inc	2014-09-29 09:23:20.000000000 +0200
@@ -100,7 +100,18 @@
   if ($user->uid) {
     // The existing user is already logged in.
     if ($user->uid == $uid) {
-      drupal_set_message(t('You are logged in as %user. <a href="!user_edit">Change your password.</a>', array('%user' => $user->name, '!user_edit' => url("user/$user->uid/edit"))));
+      //drupal_set_message(t('You are logged in as %user. <a href="!user_edit">Change your password.</a>', array('%user' => $user->name, '!user_edit' => url("user/$user->uid/edit"))));
+      // The user is already logged in, but is shown a message inviting them to
+      // change their password, since that may be their intent. Let the user's
+      // password be changed without the current password check, as even though
+      // they are logged in, they may not know it.
+      $token = drupal_random_key();
+      $_SESSION['pass_reset_' . $user->uid] = $token;
+      drupal_set_message(t('You are logged in as %user. <a href="!user_edit">Change your password.</a>', array(
+        '%user' => $user->name,
+        '!user_edit' => url("user/$user->uid/edit", array('query' => array('pass-reset-token' => $token))),
+      )));
+
     }
     // A different user is already logged in on the computer.
     else {
