--- sites/all/modules/contributed/chr/chr.module	2014-10-17 14:29:42.338714671 +0200
+++ sites/all/modules/contributed/chr/chr.module	2014-10-17 14:33:54.994184777 +0200
@@ -444,6 +444,13 @@
       curl_setopt($ch, CURLOPT_HTTPHEADER, array('X-HTTP-Method-Override: PUT'));
       $valid_method = TRUE;
     break;
+    // FPFIS fix jira REST remoting
+    // https://answers.atlassian.com/questions/72475/jira-update-custom-fields-using-rest-php-and-curl
+    case 'PUT_JIRA':
+      curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
+      curl_setopt($ch, CURLOPT_POSTFIELDS, $options['data']);
+      $valid_method = TRUE;
+      break;
     case 'GET':
       curl_setopt($ch, CURLOPT_HTTPGET, TRUE);
       $valid_method = TRUE;
