--- sites/all/modules/contributed/feeds/plugins/FeedsProcessor.inc
+++ sites/all/modules/contributed/feeds/plugins/FeedsProcessor.inc
@@ -535,7 +535,6 @@ abstract class FeedsProcessor extends FeedsPlugin {
   protected function existingEntityId(FeedsSource $source, FeedsParserResult $result) {
     $query = db_select('feeds_item')
       ->fields('feeds_item', array('entity_id'))
-      ->condition('feed_nid', $source->feed_nid)
       ->condition('entity_type', $this->entityType())
       ->condition('id', $source->id);
 
