--- sites/all/modules/contributed/media/includes/media.filter.inc	2013-08-17 00:44:16.000000000 +0200
+++ sites/all/modules/contributed/media/includes/media.filter.inc	2014-08-20 13:42:16.000000000 +0200
@@ -634,9 +634,7 @@
   field_attach_form('file', $file, $form['options']['fields'], $form_state);
   $instance = field_info_instances('file', $file->type);
   foreach ($instance as $field_name => $field_value) {
-    if (isset($instance[$field_name]['settings']) && isset($instance[$field_name]['settings']['wysiwyg_override']) && !$instance[$field_name]['settings']['wysiwyg_override']) {
-      unset($form['options']['fields'][$field_name]);
-    }
+    $form['options']['fields'][$field_name]['#access'] = !empty($instance[$field_name]['settings']['wysiwyg_override']);
   }
 
   // Similar to a form_alter, but we want this to run first so that
