--- sites/all/modules/contributed/media_vimeo/includes/MediaVimeoStreamWrapper.inc	2014-03-26 09:36:04.000000000 +0100
+++ sites/all/modules/contributed/media_vimeo/includes/MediaVimeoStreamWrapper.inc	2014-06-16 17:26:02.403040483 +0200
@@ -29,7 +29,7 @@
     $oembed_url = url('http://vimeo.com/api/oembed.json', array('query' => array('url' => $external_url)));
     $response = drupal_http_request($oembed_url);
 
-    if (!isset($response->error)) {
+    if(strlen(trim($response->error)) == 0) {
       $data = drupal_json_decode($response->data);
       return $data['thumbnail_url'];
     }
