--- sites/all/modules/contributed/media_flickr/includes/media_flickr.variables.inc
+++ sites/all/modules/contributed/media_flickr/includes/media_flickr.variables.inc
@@ -14,7 +14,7 @@
  */
 define('MEDIA_FLICKR_NAMESPACE', 'media_flickr__');

-define('MEDIA_FLICKR_RESTPOINT', 'http://api.flickr.com/services/rest/');
+define('MEDIA_FLICKR_RESTPOINT', 'https://api.flickr.com/services/rest/');
 define('MEDIA_FLICKR_API_APPLICATION_URL', 'http://www.flickr.com/services/api/keys');
 define('MEDIA_FLICKR_MAIN_URL', 'http://www.flickr.com/');

