--- sites/all/modules/features/custom/multisite_twitterblock/multisite_twitterblock.module	Wed Jul  9 13:25:40 2014
+++ sites/all/modules/features/custom/multisite_twitterblock/multisite_twitterblock.module	Thu Sep 11 16:11:54 2014
@@ -368,16 +368,22 @@
  * Implements hook_form_BASE_FORM_ID_alter().
  */
 function multisite_twitterblock_form_taxonomy_form_term_alter(&$form, $form_state) {
-  $machine_name = isset($form['#term']['vocabulary_machine_name']) ? $form['#term']['vocabulary_machine_name'] : NULL;
+
+  if (is_object($form['#term'])) {
+    $machine_name = $form['#term']->vocabulary_machine_name;
+  } else {
+    $machine_name = isset($form['#term']['vocabulary_machine_name']) ? $form['#term']['vocabulary_machine_name'] : NULL;
+  }
+
   if (!multisite_twitterblock_check_availability($machine_name, MULTISITE_TWITTERBLOCK_VOCABULARY_LIST)) {
     return;
   }
-
+
   $settings = NULL;
   if (empty($form['#term']['tid'])) {
     $settings = _multisite_twitterblock_load_settings($form['#term'], NULL, MULTISITE_TWITTERBLOCK_ENTITY_TAXONOMY_TERM);
   }
-
+
   _multisite_twitterblock_entity_subform($form, $settings);
 }

@@ -694,7 +700,7 @@
     '#type' => 'textfield',
     '#title' => t('Hashtags'),
     '#default_value' => $default_values['incl_hashtags'],
-    '#description' => t('Comma seperated list of hashtags: search for tweets containing all specified hashtags'),
+    '#description' => t('List of hashtags: search for tweets with the specified hashtags. You can divide them with comma, AND, OR'),
   );

   $form['smk-twitter_search'][$prefix . 'incl_profiles'] = array(
