--- sites/all/modules/contributed/password_policy/password_policy.js	2013-02-27 11:05:13.000000000 +0100
+++ sites/all/modules/contributed/password_policy/password_policy.js	2013-02-27 11:10:31.000000000 +0100
@@ -9,5 +9,5 @@
     pw_status = {
       strength: 0,
       message: '',
-      indicatorText: '',
+      indicatorText: ''
     }
@@ -37,5 +37,5 @@
     Drupal.evaluatePasswordStrength = function (password, translate) {
       return pw_status;
     };
-  },
+  }
 };
