--- profiles/standard/standard.install	2012-10-12 15:40:19.246713436 +0200
+++ profiles/standard/standard.install	2012-10-12 15:48:57.734995783 +0200
@@ -216,7 +216,7 @@
       'type' => 'article',
       'name' => st('Article'),
       'base' => 'node_content',
-      'description' => st('Use <em>articles</em> for time-sensitive content like news, press releases or blog posts.'),
+      'description' => st('Use <em>articles</em> for time-sensitive content.'),
       'custom' => 1,
       'modified' => 1,
       'locked' => 0,
