--- sites/all/modules/contributed/sweaver/sweaver.module	2014-09-08 12:18:26.685813121 +0200
+++ sites/all/modules/contributed/sweaver/sweaver.module	2014-09-08 12:16:03.070101271 +0200
@@ -423,6 +423,8 @@
         'group' => CSS_THEME,
       );
       $css_path = 'public://sweaver/sweaver_' . $sweaver->get_theme_key() . '_' . $sweaver_style->style_id . '_' . $sweaver_style->type . '.css';
+      $css_path = file_create_url($css_path);    
+ 
       drupal_add_css($css_path, $settings);
 
       foreach (array_keys($sweaver->get_plugins_registry(TRUE)) as $plugin_name) {
