--- sites/all/modules/contributed/sweaver/plugins/sweaver_plugin_editor.js	Sat Apr 07 17:10:26 2012
+++ sites/all/modules/contributed/sweaver/plugins/sweaver_plugin_editor.js	Fri Nov 21 12:36:43 2014
@@ -513,7 +513,7 @@
   });
 
   // Update css when something (that is not checkbox or radio button) is changed in the form.
-  $("#sweaver_plugin_editor input[id^=edit-], #sweaver_plugin_editor select[id^=edit-]").live('change', function(){
+  $("#sweaver_plugin_editor input[id^=edit-], #sweaver_plugin_editor select[id^=edit-]").on('change', function(){
     if (Drupal.Sweaver.updateMode) {
       // Is this a file input ?
       if ($(this).attr('name').match('^files\[[a-zA-Z0-9_-]+\]')){
@@ -548,7 +548,7 @@
     }
   });
   
-  $('#sweaver_plugin_editor .form-managed-file input[type=submit]').live('mouseover', function(){
+  $('#sweaver_plugin_editor .form-managed-file input[type=submit]').on('mouseover', function(){
     if(!$(this).hasClass('event_added')){
       $(this).addClass('event_added');
       $(this).bind('mousedown', function(){
