--- modules/system/maintenance-page.tpl.php 2013-04-10 10:19:38.000000000 +0200
+++ modules/system/maintenance-page.tpl.php	2013-04-10 10:19:45.000000000 +0200
@@ -29,7 +29,7 @@
       <div id="logo-title">
 
         <?php if (!empty($logo)): ?>
-          <a href="<?php print $base_path; ?>" title="<?php print t('Home'); ?>" rel="home" id="logo">
+          <a href="<?php print filter_xss($base_path); ?>" title="<?php print t('Home'); ?>" rel="home" id="logo">
             <img src="<?php print $logo; ?>" alt="<?php print t('Home'); ?>" />
           </a>
         <?php endif; ?>
