--- sites/all/modules/contributed/entity/theme/entity.tpl.php	2013-11-28 13:05:33.879734014 +0100
+++ sites/all/modules/contributed/entity/theme/entity.tpl.php	2013-11-28 15:05:21.000000000 +0100
@@ -33,7 +33,7 @@
   <?php if (!$page): ?>
     <h2<?php print $title_attributes; ?>>
       <?php if ($url): ?>
-        <a href="<?php print $url; ?>"><?php print $title; ?></a>
+        <a href="<?php print filter_xss($url); ?>"><?php print $title; ?></a>
       <?php else: ?>
         <?php print $title; ?>
       <?php endif; ?>
