--- modules/aggregator/aggregator-feed-source.tpl.php	2013-04-15 15:26:23.054930141 +0200
+++ modules/aggregator/aggregator-feed-source.tpl.php	2013-04-16 09:03:27.000000000 +0200
@@ -28,7 +28,7 @@
     <?php print $source_description; ?>
   </div>
   <div class="feed-url">
-    <em><?php print t('URL:'); ?></em> <a href="<?php print $source_url; ?>"><?php print $source_url; ?></a>
+    <em><?php print t('URL:'); ?></em> <a href="<?php print filter_xss($source_url); ?>"><?php print filter_xss($source_url); ?></a>
   </div>
   <div class="feed-updated">
     <em><?php print t('Updated:'); ?></em> <?php print $last_checked; ?>
