--- modules/node/node.tpl.php	2013-04-10 14:47:49.801173977 +0200
+++ modules/node/node.tpl.php	2013-04-11 08:26:35.000000000 +0200
@@ -86,7 +86,7 @@
 
   <?php print render($title_prefix); ?>
   <?php if (!$page): ?>
-    <h2<?php print $title_attributes; ?>><a href="<?php print $node_url; ?>"><?php print $title; ?></a></h2>
+    <h2<?php print $title_attributes; ?>><a href="<?php print filter_xss($node_url); ?>"><?php print filter_xss($title); ?></a></h2>
   <?php endif; ?>
   <?php print render($title_suffix); ?>
 
