index d874492..2b5d6bf 100644
--- og.install
+++ og.install
@@ -1187,3 +1187,10 @@ function og_update_7204() {
 function og_update_7205() {
   db_add_index('og_membership', 'group_type', array('group_type'));
 }
+
+/**
+ * Remove incorrect roles.
+ */
+function og_update_7206() {
+  db_query("DELETE FROM {og_users_roles} where gid in (SELECT gid from {og_role} where gid > 0) and rid in (SELECT rid from {og_role} where gid = 0)");
+}

index 270e182..62ba3e6
--- og.module
+++ og.module
@@ -866,7 +866,7 @@ function og_entity_update($entity, $entity_type) {
   }
 
   list($id, $vid, $bundle) = entity_extract_ids($entity_type, $entity);
-  if (!empty($entity->uid) && !og_is_member($entity_type, $entity, 'user', $entity->uid)) {
+  if (!empty($entity->uid) && !og_is_member($entity_type, $id, 'user', user_load($entity->uid))) {
     // Subscribe the group manager, in case the owner changed.
     og_group($entity_type, $id, array('entity' => $entity->uid));
     // Assign roles to group manager.
