--- admin_menu.js
+++ admin_menu.js
@@ -220,7 +220,7 @@ Drupal.admin.behaviors.destination = function (context, settings, $adminMenu) {
  */
 Drupal.admin.behaviors.hover = function (context, settings, $adminMenu) {
   // Hover emulation for IE 6.
-  if ($.browser.msie && parseInt(jQuery.browser.version) == 6) {
+  if (navigator.appName == 'Microsoft Internet Explorer' && navigator.userAgent.match(/msie 6/i)) {
     $('li', $adminMenu).hover(
       function () {
         $(this).addClass('iehover');
