--- admin_menu/admin_menu.inc
+++ admin_menu/admin_menu.inc
@@ -705,7 +705,9 @@ function admin_menu_theme_settings() {
   foreach ($result as $module) {
     if (file_exists($module->filename)) {
       $info = unserialize($module->info);
-      $all_modules[$module->name] = $info['name'];
+      if (empty($info['hidden'])) {
+        $all_modules[$module->name] = $info['name'];
+      }
     }
   }
   $devel_modules = variable_get('admin_menu_devel_modules', _admin_menu_developer_modules());
