diff a/apachesolr_attachments.index.inc b/apachesolr_attachments.index.inc
--- apachesolr_attachments.index.inc	2014-07-08 15:47:59.287508791 +0200
+++ apachesolr_attachments.index.inc	2014-07-08 15:07:15.255695729 +0200
@@ -243,18 +243,3 @@
     ->condition('parent_entity_id', $parent_entity_id);
 }
 
-function apachesolr_attachments_clean_index_table() {
-  $indexer_table = apachesolr_get_indexer_table('file');
-  // Clean all entries where parent_entity_id is empty
-  db_delete($indexer_table)
-  ->condition('parent_entity_id', 0)
-  ->execute();
-  // Clean all entries from entity types that should not be indexed
-  foreach (entity_get_info() as $entity_type => $entity_info) {
-    if (empty($entity_info['apachesolr']['indexable'])) {
-       db_delete($indexer_table)
-      ->condition('parent_entity_type', $entity_type)
-      ->execute();
-    }
-  }
-}
diff  a/apachesolr_attachments.module b/apachesolr_attachments.module
--- apachesolr_attachments.module	2014-07-08 15:47:59.388509648 +0200
+++ apachesolr_attachments.module	2014-07-08 15:10:15.252234311 +0200
@@ -60,6 +60,26 @@
 }
 
 /**
+* Implements hook_cron().
+*
+* Copy of old apachesolr_attachments_clean_index_table().
+*/
+function apachesolr_attachments_cron() {
+  $indexer_table = apachesolr_get_indexer_table('file');
+  // Clean all entries where parent_entity_id is empty
+  db_delete($indexer_table)
+  ->condition('parent_entity_id', 0)
+  ->execute();
+  // Clean all entries from entity types that should not be indexed
+  foreach (entity_get_info() as $entity_type => $entity_info) {
+    if (empty($entity_info['apachesolr']['indexable'])) {
+      db_delete($indexer_table)
+      ->condition('parent_entity_type', $entity_type)
+      ->execute();
+    }
+  }
+}
+/**
  * @file
  *   Indexer for the userhook_apachesolr_entity_info_alter entities for the Apachesolr module.
  */
@@ -514,9 +534,6 @@
  * were added to the index table
  */
 function apachesolr_attachments_entity_update($entity, $type) {
-  module_load_include('inc', 'apachesolr_attachments', 'apachesolr_attachments.index');
-  apachesolr_attachments_clean_index_table();
-
   if ($type == 'file') {
     _apachesolr_attachments_update_parent_entity($entity, $type);
   }
@@ -530,10 +547,6 @@
   }
 }
 
-function apachesolr_attachments_entity_delete($entity, $type) {
-  module_load_include('inc', 'apachesolr_attachments', 'apachesolr_attachments.index');
-  apachesolr_attachments_clean_index_table();
-}
 
