--- file_entity.tpl.php	2014-01-10 13:19:46.000000000 +0100
+++ file_entity.tpl.php	2014-01-10 13:11:27.000000000 +0100
@@ -72,7 +72,7 @@
 
   <?php print render($title_prefix); ?>
   <?php if (!$page): ?>
-    <h2<?php print $title_attributes; ?>><a href="<?php print $file_url; ?>"><?php print $label; ?></a></h2>
+    <h2<?php print $title_attributes; ?>><a href="<?php print filter_xss($file_url); ?>"><?php print $label; ?></a></h2>
   <?php endif; ?>
   <?php print render($title_suffix); ?>
 
