--- js/fullcalendar.ajax.js	2014-06-26 15:36:11.205221140 +0200
+++ js/fullcalendar.ajax.js	2014-06-26 15:36:40.347390179 +0200
@@ -25,7 +25,13 @@
 
 Drupal.fullcalendar.fullcalendar.prototype.submitInit = function (settings) {
   var domId = this.dom_id.replace('.view-dom-id-', '');
-  var ajaxView = Drupal.settings.views.ajaxViews['views_dom_id:' + domId];
+  
+  try {
+    var ajaxView = Drupal.settings.views.ajaxViews['views_dom_id:' + domId];
+  } catch (e) {
+    return;
+  }
+ 
   this.tm = settings.theme ? 'ui' : 'fc';
   var $submit = this.$calendar.find('.views-exposed-form .views-submit-button');
   if (this.$calendar.find('.views-exposed-widget').length == settings.fullcalendar_fields_count + 1) {
