--- maxlength.module	2014-04-28 15:55:44.008372806 +0200
+++ maxlength.module	2014-04-28 15:56:18.633560822 +0200
@@ -168,7 +168,7 @@
  */
 function maxlength_validate_input(&$element, &$form_state) {
   // Verify that the value is not longer than #maxlength characters.
-  if (isset($element['#attributes']['maxlength'])) {
+  if (isset($element['#attributes']['maxlength']) && isset($element['#value'])) {
     // Compute the length of the text, without counting the tags, and consider
     // the "enter" characters as only one character.
     $value = filter_xss(str_replace(array("\r\n", '&nbsp;'), array(' ', ' '), $element['#value']), array());
