--- maxlength.module	2013-09-02 11:46:14.730544385 +0200
+++ maxlength.module	2013-09-02 11:46:47.657545757 +0200
@@ -185,7 +185,7 @@
  */
 function maxlength_form_alter(&$form, &$form_state, $form_id) {
   // Attach maxlength to node title
-  if (strpos($form_id, '_node_form') !== FALSE && variable_get('maxlength_js_' . $form['#node']->type, FALSE)) {
+  if (isset($form['#node']) && strpos($form_id, '_node_form') !== FALSE && variable_get('maxlength_js_' . $form['#node']->type, FALSE)) {
     if (!module_exists('title') || title_field_replacement_enabled('node', $form['#node']->type, 'title') !== TRUE) {
       $form['title']['#maxlength_js'] = TRUE;
       $form['title']['#maxlength'] = intval(variable_get('maxlength_js_' . $form['#node']->type, ''));
@@ -234,4 +234,4 @@
    if (!empty($element['#value']) && is_numeric($element['#value']) && $element['#value'] > 255) {
      form_error($element, t('Note titles can be maximum 255 characters long.'));
    }   
-}
\ No newline at end of file
+}
