Index: sites/all/modules/contributed/media/includes/media.browser.inc
===================================================================
--- includes/media.browser.inc	(revision 571)
+++ includes/media.browser.inc	(working copy)
@@ -428,5 +428,11 @@
 function media_browser_build_media_item($file) {
   $preview = media_get_thumbnail_preview($file);
   $file->preview = drupal_render($preview);
+
+  // Nested <a> elements are invalid HTML, and interfere with media browser's
+  // javascript. We cannot reliably assume the thumbnail contains compatible
+  // HTML, so we need to strip any existing links. strip_tags() is an aggressive
+  // way to accomplish this, but it's quick and effective.
+  $file->preview = strip_tags($file->preview, '<span><img><div><label>');
   $file->url = file_create_url($file->uri);
 }
