--- includes/MediaInternetVimeoHandler.inc	2014-03-26 09:36:04.000000000 +0100
+++ includes/MediaInternetVimeoHandler.inc	2014-06-16 17:26:18.018127588 +0200
@@ -67,7 +67,7 @@
     $oembed_url = url('http://vimeo.com/api/oembed.json', array('query' => array('url' => $external_url)));
     $response = drupal_http_request($oembed_url);
 
-    if (!isset($response->error)) {
+    if(strlen(trim($response->error)) == 0) {
       return drupal_json_decode($response->data);
     }
     else {
