--- quicktabs.classes.inc	2013-10-02 18:03:07.000000000 +0200
+++ quicktabs.classes.inc	2014-07-30 12:10:05.000000000 +0200
@@ -244,7 +244,12 @@
   public function getActiveTab() {
     $active_tab = isset($this->settings['default_tab']) ? $this->settings['default_tab'] : key($this->contents);
     $active_tab = isset($_GET['qt-' . $this->name]) ? $_GET['qt-' . $this->name] : $active_tab;
-    $active_tab = (isset($active_tab) && isset($this->contents[$active_tab])) ? $active_tab : QUICKTABS_DELTA_NONE;
+    if(isset($this->settings['default_tab']) && $this->settings['default_tab'] == QUICKTABS_DELTA_NONE) {
+      $active_tab = (isset($active_tab) && isset($this->contents[$active_tab])) ? $active_tab : QUICKTABS_DELTA_NONE;
+    }
+    else {
+      $active_tab = (isset($active_tab) && isset($this->contents[$active_tab])) ? $active_tab : key($this->contents);
+    }
     return $active_tab;
   }
 }
