--- theme/views-view-summary.tpl.php	2013-04-15 12:54:54.553179964 +0200
+++ theme/views-view-summary.tpl.php	2013-04-15 13:16:53.000000000 +0200
@@ -10,7 +10,7 @@
 <div class="item-list">
   <ul class="views-summary">
   <?php foreach ($rows as $id => $row): ?>
-    <li><a href="<?php print $row->url; ?>"<?php print !empty($row_classes[$id]) ? ' class="'. $row_classes[$id] .'"' : ''; ?>><?php print $row->link; ?></a>
+    <li><a href="<?php print filter_xss($row->url); ?>"<?php print !empty($row_classes[$id]) ? ' class="'. $row_classes[$id] .'"' : ''; ?>><?php print $row->link; ?></a>
       <?php if (!empty($options['count'])): ?>
         (<?php print $row->count?>)
       <?php endif; ?>
