--- includes/view.inc
+++ includes/view.inc
@@ -2076,25 +2076,26 @@ class view extends views_db_object {
   }
 
   /**
-   * Find and initialize the localizer plugin.
+   * Find and initialize the localization plugin.
    */
   function init_localization() {
-    if (isset($this->localization_plugin) && is_object($this->localization_plugin)) {
-      return TRUE;
-    }
+    // If the translate attribute isn't set, init the localization plugin.
+    if (!isset($this->localization_plugin->translate)) {
+      $this->localization_plugin = views_get_plugin('localization', views_get_localization_plugin());
 
-    $this->localization_plugin = views_get_plugin('localization', views_get_localization_plugin());
+      // If the plugin is still not set, turn off all localization by using the
+      // views_plugin_localization_none plugin. This plugin has the translate
+      // property set to FALSE, signifying localization should not occur.
+      if (empty($this->localization_plugin)) {
+        $this->localization_plugin = views_get_plugin('localization', 'none');
+      }
 
-    if (empty($this->localization_plugin)) {
-      $this->localization_plugin = views_get_plugin('localization', 'none');
-      return FALSE;
+      // Init the plugin.
+      $this->localization_plugin->init($this);
     }
 
-    /**
-    * Figure out whether there should be options.
-    */
-    $this->localization_plugin->init($this);
-
+    // Return the value of the translate property. This is set to FALSE if
+    // localization is off.
     return $this->localization_plugin->translate;
   }

