--- workbench_access.module	2014-09-17 15:07:29.000000000 +0200
+++ workbench_access.module	2014-09-17 15:10:05.000000000 +0200
@@ -1202,6 +1202,8 @@
  *   An array of items within the given access scheme.
  */
 function workbench_access_build_tree(&$tree, $sections = NULL, $depth = -1) {
+  if(empty($tree))
+    return;
   static $max_depth;
   if (!isset($max_depth)) {
     $max_depth = 0;
@@ -1440,6 +1442,8 @@
  */
 function workbench_access_allow_form_alter() {
   $active = workbench_access_get_active_tree();
+  if(!is_array($active['access_scheme']['access_type_id']))
+    return FALSE;
   $ids = array_filter($active['access_scheme']['access_type_id']);
   if (!empty($ids)) {
     return TRUE;
