diff --git a/workbench_moderation.module b/workbench_moderation.module
index 9a9b04d..8803502 100644
--- workbench_moderation.module
+++ workbench_moderation.module
@@ -1575,6 +1575,16 @@ function workbench_moderation_store($node) {
   $taxonomy_index_flag = &drupal_static('taxonomy_field_update', array());
   unset($taxonomy_index_flag[$node->nid]);
 
+  // Ensure we do not have field translations belonging to a draft revision in
+  // the field data tables.
+  $empty_values = array_fill_keys(array_keys(language_list()), array());
+  foreach (field_info_instances('node', $live_revision->type) as $field_name => $instance) {
+    $field = field_info_field($field_name);
+    if (!empty($live_revision->{$field_name}) && field_is_translatable('node', $field)) {
+      $live_revision->{$field_name} += $empty_values;
+    }
+  }
+
   // Save the node.
   node_save($live_revision);
 }
