--- workbench_moderation.module 2014-02-26 13:47:25.001581317 +0100
+++ workbench_moderation.module 2014-02-26 13:46:58.612427976 +0100
@@ -697,8 +697,6 @@
   // Apply moderation changes if this is a new revision or if the moderation
   // state has changed.
   if (!empty($node->revision) || $node->workbench_moderation_state_current != $node->workbench_moderation_state_new) {
-    // Update attached fields.
-    field_attach_update('node', $node);
     // Moderate the node.
     workbench_moderation_moderate($node, $node->workbench_moderation_state_new);
   }
@@ -1574,6 +1572,9 @@
   $taxonomy_index_flag = &drupal_static('taxonomy_field_update', array());
   unset($taxonomy_index_flag[$node->nid]);

+  // Set original node for current vid
+  $live_revision->original = node_load($live_revision->nid, $live_revision->vid, TRUE);
+
   // Save the node.
   node_save($live_revision);
 }
