--- theme/views-view-summary-unformatted.tpl.php	2013-04-15 12:54:54.553179964 +0200
+++ theme/views-view-summary-unformatted.tpl.php	2013-04-15 13:13:51.000000000 +0200
@@ -12,7 +12,7 @@
 <?php foreach ($rows as $id => $row): ?>
   <?php print (!empty($options['inline']) ? '<span' : '<div') . ' class="views-summary views-summary-unformatted">'; ?>
     <?php if (!empty($row->separator)) { print $row->separator; } ?>
-    <a href="<?php print $row->url; ?>"<?php print !empty($row_classes[$id]) ? ' class="' . $row_classes[$id] . '"' : ''; ?>><?php print $row->link; ?></a>
+    <a href="<?php print filter_xss($row->url); ?>"<?php print !empty($row_classes[$id]) ? ' class="' . $row_classes[$id] . '"' : ''; ?>><?php print filter_xss($row->link); ?></a>
     <?php if (!empty($options['count'])): ?>
       (<?php print $row->count; ?>)
     <?php endif; ?>
