--- misc/ajax.js	2014-08-14 15:22:49.803356572 +0200
+++ misc/ajax.js	2014-08-14 15:23:06.191439888 +0200
@@ -131,7 +131,7 @@
   //      E.g.: ?q=path/nojs&destination=foobar
   // 5. /nojs# - Followed by a fragment.
   //      E.g.: path/nojs#myfragment
-  this.url = element_settings.url.replace(/\/nojs(\/|$|\?|&|#)/g, '/ajax$1');
+  this.url = element_settings.url.replace(/\/nojs(\/|$|\?|&|#|_)/g, '/ajax$1');
   // If the 'nojs' version of the URL is trusted, also trust the 'ajax' version.
   if (Drupal.settings.urlIsAjaxTrusted[element_settings.url]) {
     Drupal.settings.urlIsAjaxTrusted[this.url] = true;
