diff --git a/Drupal_Apache_Solr_Service.php b/Drupal_Apache_Solr_Service.php
index e67f483..1bddfe0 100644
--- a/Drupal_Apache_Solr_Service.php
+++ b/Drupal_Apache_Solr_Service.php
@@ -552,6 +552,9 @@ class DrupalApacheSolrService implements DrupalApacheSolrServiceInterface {
       $options['data'] = NULL;
     }
 
+    // @todo: Remove after investigation of 500 errors - NEXTEUROPA-11356.
+    // Warning 'timeout' is hardcoded for all of HTTP SOLR requests.
+    $options['timeout'] = 5;
     $result = drupal_http_request($url, $options);
 
     if (!isset($result->code) || $result->code < 0 || isset($result->error)) {
