--- apachesolr.module	2013-08-12 18:09:05.917742517 +0200
+++ apachesolr.module	2013-08-12 18:10:15.884814584 +0200
@@ -1876,9 +1876,11 @@
           $supported = apachesolr_get_index_bundles($env['env_id'], $entity_type);
           // For each bundle in drupal, compare to the supported apachesolr
           // bundles and enable where possible
-          foreach (array_keys($info['bundles']) as $bundle) {
-            if (in_array($bundle, $supported)) {
-              $entity_info[$entity_type]['bundles'][$bundle]['apachesolr']['index'] = TRUE;
+          if(isset($info['bundles'])) {
+            foreach (array_keys($info['bundles']) as $bundle) {
+              if (in_array($bundle, $supported)) {
+                $entity_info[$entity_type]['bundles'][$bundle]['apachesolr']['index'] = TRUE;
+              }
             }
           }
         }
