--- apachesolr.index.inc	2013-11-14 16:49:29.000000000 +0100
+++ apachesolr.index.inc	2014-08-20 09:15:09.000000000 +0200
@@ -1218,16 +1218,22 @@
     // and one for the end date.
     foreach ($values as $value) {
       if ($date = date_create($value['value'], $tz)) {
-        $index_value = apachesolr_date_iso($date->format('U'));
-        $fields[] = array(
-          'key' => $index_key,
-          'value' => $index_value,
-        );
-      }
+      // Add one second when creating the date to avoid problems with date
+      // fields that have low granularity appearing on the boundary between
+      // two ranges.
+      $index_value = apachesolr_date_iso($date->format('U') + 1);
+      $fields[] = array(
+        'key' => $index_key,
+        'value' => $index_value,
+      );
+    }
 
       if (isset($value['value2'])) {
         if ($date = date_create($value['value2'], $tz)) {
-          $index_value = apachesolr_date_iso($date->format('U'));
+          // Add one second when creating the date to avoid problems with date
+          // fields that have low granularity appearing on the boundary between
+          // two ranges.
+          $index_value = apachesolr_date_iso($date->format('U') + 1);
           $fields[] = array(
             // The value2 element is the end date. Therefore it gets indexed
             // into its own Solr field.
@@ -1261,14 +1267,20 @@
 
     foreach ($values as $value) {
       if (isset($value['value']) && $value['value'] != 0) {
-        $index_value = apachesolr_date_iso($value['value']);
+        // Add one second when creating the date to avoid problems with date
+        // fields that have low granularity appearing on the boundary between
+        // two ranges.
+        $index_value = apachesolr_date_iso($value['value'] + 1);
         $fields[] = array(
           'key' => $index_key,
           'value' => $index_value,
         );
       }
       if (isset($value['value2']) && $value['value'] != 0) {
-        $index_value = apachesolr_date_iso($value['value2']);
+        // Add one second when creating the date to avoid problems with date
+        // fields that have low granularity appearing on the boundary between
+        // two ranges.
+        $index_value = apachesolr_date_iso($value['value2'] + 1);
         $fields[] = array(
           // The value2 element is the end date. Therefore it gets indexed
           // into its own Solr field.
