diff --git a/entity/tmgmt.entity.job_item.inc b/entity/tmgmt.entity.job_item.inc
index b1e7114..8bc4bfd 100644
--- a/entity/tmgmt.entity.job_item.inc
+++ b/entity/tmgmt.entity.job_item.inc
@@ -401,21 +401,180 @@ class TMGMTJobItem extends Entity {
 
   /**
    * Sets the state of the job item to 'needs review'.
+   *
+   * @param string $message
+   * @param array $variables
+   * @param string $type
+   * @return int
+   * @throws TMGMTException
    */
-  public function needsReview($message = NULL, $variables = array(), $type = 'status') {
-    if (!isset($message)) {
+  public function needsReview($message = NULL, array $variables = array(), $type = 'status') {
+    if ($message === NULL) {
       $uri = $this->getSourceUri();
       $message = 'The translation for !source needs to be reviewed.';
       $variables = array('!source' => l($this->getSourceLabel(), $uri['path']));
     }
     $return = $this->setState(TMGMT_JOB_ITEM_STATE_REVIEW, $message, $variables, $type);
-    // Auto accept the trganslation if the translator is configured for it.
+    // Auto accept the translation if the translator is configured for it.
     if ($this->getTranslator()->getSetting('auto_accept')) {
+      $this->validateFieldMaxLength($this->data);
       $this->acceptTranslation();
     }
+
     return $return;
   }
 
+    /**
+     * Validate each item field against its default max length allowed.
+     * This way it avoids exception while saving on the database.
+     * "File import failed with the following message: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'field_teste_value'"
+     *
+     * @param array $data
+     *   A structured data array.
+     * @throws TMGMTException
+     */
+    private function validateFieldMaxLength(array $data) {
+
+      foreach ($data as $field_name => $field_data) {
+        $field_info = field_info_field($field_name);
+
+        // Skip fields type "text_with_summary" which the column type is "longtext/bg", such as body field..
+        if ($field_info === NULL) {
+          continue;
+        }
+
+        $max_allowed_length = $this->fieldSettingsLength($field_info);
+
+        // Skip the longtext, with a maximum length of 4 GiB (maximum characters depending on the encoding).
+        // Represented as 'bg', which normally found in body field, text areas, etc.
+        if (!is_numeric($max_allowed_length)) {
+          continue;
+        }
+
+        $translation_length = $this->fieldDataLength($field_info, $field_data);
+
+        if ($max_allowed_length >= $translation_length) {
+          continue;
+        }
+
+        // By setting the state it avoids a second try to be finished with a false successful message "Successfully imported file."
+        $this->setState(TMGMT_JOB_ITEM_STATE_ACTIVE, 'Failed to validate file, import aborted.', array(), 'error');
+        throw new TMGMTException(t('The data on %field_name should be equal or less than %max_allowed_length characters, found %translation_length characters!', array('%field_name' => $field_name, '%max_allowed_length' => $max_allowed_length, '%translation_length' => $translation_length)));
+      }
+    }
+
+    /**
+     * Manage the data length from imported file with translated data items.
+     *
+     * @param array $field_info
+     *   Field info object.
+     * @param array $field_data
+     *   Field data structure as array.
+     * @return int
+     *   The data length, zero it it is empty or null if it is not set.
+     */
+    private function fieldDataLength(array &$field_info, array &$field_data) {
+      switch ($field_info['type']) {
+        case 'link_field':
+          $data = $this->linkData($field_data);
+          break;
+        case 'text_long':
+        case 'text_with_summary':
+          $data = $this->textSummaryLongData($field_data);
+          break;
+        default:
+          $data = isset($field_data[0]['value']['#translation']['#text']) ? $field_data[0]['value']['#translation']['#text'] : NULL;
+          break;
+      }
+
+      return drupal_strlen($data);
+    }
+
+    /**
+     * Get the translation data from the text_with_summary and or text_long field type.
+     *
+     * @param array $data
+     *   The data elements from translation file.
+     * @param string $column
+     *   The column can be value and or summary.
+     *
+     * @return mixed
+     */
+    private function textSummaryLongData(array &$data, $column = 'value') {
+      return isset($data[0][$column]['#translation']['#text']) ? $data[0][$column]['#translation']['#text'] : NULL;
+    }
+
+    /**
+     * Manage field settings length.
+     *
+     * @param array $field_info.
+     *  The object field elements.
+     * @return int|mixed|null
+     */
+    private function fieldSettingsLength(array &$field_info) {
+      switch ($field_info['type']) {
+        case 'link_field':
+          $length = $this->linkTitleLengthSettings($field_info);
+          break;
+
+        case 'text_long':
+        case 'text_with_summary':
+          $length = $this->textSummaryLongLengthSettings($field_info);
+          break;
+
+        default:
+          $length = $field_info['settings']['max_length'];
+          break;
+      }
+
+      return $length;
+    }
+
+    /**
+     * Get text_with_summary and text_long field type length.
+     *
+     * @param array $field_info
+     *   The object field elements.
+     * @param string $column
+     *   The column name: {value, summary and format}.
+     * @return mixed
+     */
+    public function textSummaryLongLengthSettings(array &$field_info, $column = 'value') {
+      return isset($field_info['columns'][$column]['size']) ? $field_info['columns'][$column]['size'] : NULL;
+    }
+
+    /**
+     * Get the custom length on link field type from settings.
+     * The length which was set while creating the bundle, otherwise it is 128 characters by default.
+     *
+     * @param array $field_info
+     *   The object field elements.
+     * @param string $entity_type
+     *   The entity type for the instance.
+     * @return int|null
+     *  The max number of characters allowed.
+     */
+    private function linkTitleLengthSettings(array &$field_info, $entity_type = 'node') {
+      // May have many bundles, in case of the same field being attached to many content type, however the settings are the same, so just pick up the first one.
+      $bundle = isset($field_info['bundles'][$entity_type][0]) ? $field_info['bundles'][$entity_type][0] : NULL;
+      $field_instance = field_info_instance($entity_type, $field_info['field_name'], $bundle);
+
+      return isset($field_instance['settings']['title_maxlength']) ? $field_instance['settings']['title_maxlength'] : NULL;
+    }
+
+    /**
+     * Get the translation data from link field type.
+     *
+     * @param array $data
+     *   The data elements from translation file.
+     * @param string $column
+     *   The element name, such as title and summary.
+     * @return mixed
+     */
+    private function linkData(array &$data, $column = 'title') {
+      return isset($data[0][$column]['#translation']['#text']) ? $data[0][$column]['#translation']['#text'] : NULL;
+    }
+
   /**
    * Sets the state of the job item to 'accepted'.
    */
