diff --git a/multisiteplayground_master_production.2.1/includes/locale.inc b/multisiteplayground_master_production.2.1/includes/locale.inc
index 82c55e5..e086f0d 100644
--- includes/locale.inc
+++ includes/locale.inc
@@ -1485,6 +1485,10 @@ function _locale_parse_js_file($filepath) {
   $parsed_url = drupal_parse_url($filepath);
   $filepath = $parsed_url['path'];
   // Load the JavaScript file.
+  if(filter_var($filepath, FILTER_VALIDATE_URL)) {
+    watchdog('remote denied', 'Remote loading of '.$filepath.' trough _locale_parse_js_file prevented');
+    return true;
+  }
   $file = file_get_contents($filepath);
 
   // Match all calls to Drupal.t() in an array.

