diff --git a/includes/context.inc b/includes/context.inc
index 1f9c1e4..2ed3f26 100644
--- a/includes/context.inc
+++ b/includes/context.inc
@@ -42,7 +42,7 @@ class ctools_context {
   var $restrictions = array();
   var $empty = FALSE;
 
-  function ctools_context($type = 'none', $data = NULL) {
+  function __construct($type = 'none', $data = NULL) {
     $this->type  = $type;
     $this->data  = $data;
     $this->title = t('Unknown context');
@@ -119,7 +119,7 @@ class ctools_context_required {
    * @param ...
    *   One or more keywords to use for matching which contexts are allowed.
    */
-  function ctools_context_required($title) {
+  function __construct($title) {
     $args = func_get_args();
     $this->title = array_shift($args);
 
@@ -204,9 +204,8 @@ class ctools_context_required {
  */
 class ctools_context_optional extends ctools_context_required {
   var $required = FALSE;
-  function ctools_context_optional() {
-    $args = func_get_args();
-    call_user_func_array(array($this, 'ctools_context_required'), $args);
+  function __construct() {
+    parent::__construct(NULL);
   }
 
   /**
diff --git a/includes/math-expr.inc b/includes/math-expr.inc
index eeb184d..3105ec5 100644
--- a/includes/math-expr.inc
+++ b/includes/math-expr.inc
@@ -99,7 +99,7 @@ class ctools_math_expr {
         'sqrt','abs','ln','log',
         'time', 'ceil', 'floor', 'min', 'max', 'round');
 
-    function ctools_math_expr() {
+    function __construct() {
         // make the variables a little more accurate
         $this->v['pi'] = pi();
         $this->v['e'] = exp(1);
