--- sites/default/default.settings.php	2015-10-21 14:34:16.000000000 +0200
+++ sites/default/default.settings.php	2015-10-21 14:41:51.000000000 +0200
@@ -1,5 +1,10 @@
 <?php
 
+$local_settings = getenv('DRUPAL_LOCAL_SETTINGS_LOCATION');
+if (file_exists($local_settings)){
+  include($local_settings);
+}
+
 /**
  * @file
  * Drupal site-specific configuration file.
@@ -584,3 +589,8 @@
  * Remove the leading hash sign to enable.
  */
 # $conf['theme_debug'] = TRUE;
+
+$local_settings_post = str_replace('.php', '.post.php', $local_settings);
+if (file_exists($local_settings_post)){
+ include($local_settings_post);
+}
