diff --git a/includes/common.inc b/includes/common.inc
index c6303ef..87fcf1d 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -4206,6 +4206,16 @@ function drupal_add_js($data = NULL, $options = NULL) {
     $jquery_added = FALSE;
   }
 
+  // If no type is specified and the URL is external, force the type to external.
+  if (is_string($data) && url_is_external($data)) {
+    if (is_array($options) && !array_key_exists('type', $options)) {
+      $options['type'] = 'external';
+    }
+    elseif (is_null($options)) {
+      $options = 'external';
+    }
+  }
+
   // Construct the options, taking the defaults into consideration.
   if (isset($options)) {
     if (!is_array($options)) {
