--- js/eu_cookie_compliance.js.original
+++ js/eu_cookie_compliance.js
@@ -81,7 +81,7 @@
   }
 
   Drupal.eu_cookie_compliance.attachEvents = function() {
-	var clicking_confirms = Drupal.settings.eu_cookie_compliance.popup_clicking_confirmation;
+    var clicking_confirms = Drupal.settings.eu_cookie_compliance.popup_clicking_confirmation;
     var agreed_enabled = Drupal.settings.eu_cookie_compliance.popup_agreed_enabled;
     $('.find-more-button').click(function(){
       window.open(Drupal.settings.eu_cookie_compliance.popup_link);
@@ -103,18 +103,9 @@
   }
 
   Drupal.eu_cookie_compliance.getCurrentStatus = function() {
-    var search = 'cookie-agreed-'+Drupal.settings.eu_cookie_compliance.popup_language+'=';
-    var offset = document.cookie.indexOf(search);
-    if (offset < 0) {
-      return 0;
-    }
-    offset += search.length;
-    var end = document.cookie.indexOf(';', offset);
-    if (end == -1) {
-      end = document.cookie.length;
-    }
-    var value = document.cookie.substring(offset, end);
-    return parseInt(value);
+    name = 'cookie-agreed';
+    value = Drupal.eu_cookie_compliance.getCookie(name);
+    return value;
   }
 
   Drupal.eu_cookie_compliance.changeStatus = function(value) {
@@ -147,7 +138,7 @@
   Drupal.eu_cookie_compliance.setStatus = function(status) {
     var date = new Date();
     date.setDate(date.getDate() + 100);
-    document.cookie = "cookie-agreed-"+Drupal.settings.eu_cookie_compliance.popup_language + "="+status+";expires=" + date.toUTCString() + ";path=" + Drupal.settings.basePath;
+    document.cookie = "cookie-agreed=" + status + ";expires=" + date.toUTCString() + ";path=" + Drupal.settings.basePath;
   }
 
   Drupal.eu_cookie_compliance.hasAgreed = function() {
@@ -157,6 +148,26 @@
     }
     return false;
   }
+
+  /*
+  Javascript Standard Cookie Getter
+  */
+ Drupal.eu_cookie_compliance.getCookie = function(name) {
+  var search = name + '=';
+  var returnCookie = '';
+  if (document.cookie.length > 0) {
+    offset = document.cookie.indexOf(search);
+    if (offset != -1) {
+      offset += search.length;
+      var end = document.cookie.indexOf(';', offset);
+      if (end == -1) {
+        end = document.cookie.length;
+      }
+      returnCookie = decodeURIComponent(document.cookie.substring(offset, end).replace(/\+/g, '%20'));
+    }
+  }
+ return returnCookie;
+ }
   
   Drupal.eu_cookie_compliance.cookiesEnabled = function() {
     var cookieEnabled = (navigator.cookieEnabled) ? true : false;
