--- feature_set.admin.inc	2015-02-25 14:00:55.000000000 +0100
+++ feature_set.admin.inc	2015-07-01 15:23:39.000000000 +0200
@@ -33,6 +33,7 @@
             'featureset' => $details['title'],
             'description' => $details['description'],
             'icon' => empty($details['icon']) ? '' : $details['icon'],
+            'category' => empty($details['category']) ? '' : $details['category'],
           ),
         );
 
--- feature_set.module	2015-02-25 14:00:55.000000000 +0100
+++ feature_set.module	2017-04-18 13:47:20.000000000 +0200
@@ -268,6 +268,43 @@
     'feature_set_admin_form' => array(
       'file' => 'feature_set.admin.inc',
       'render element' => 'form',
+	  'template' => 'feature-set-admin-form',
     ),
   );
 }
+
+/**
+ * Implements template_preprocess().
+ */
+function feature_set_preprocess_feature_set_admin_form(&$variables) {
+  if (isset($variables['form']['#form_id']) && $variables['form']['#form_id'] == 'feature_set_admin_form') {
+    $feature_set_category = array();
+    foreach ($variables['form'] as $key => $item) {
+      $variables['feature_set_row'][$key] = $item;
+
+      // Get category if available.
+      if (is_array($item) && isset($item['#featuresetinfo'])) {
+        if (empty($item['#featuresetinfo']['category'])) {
+          $feature_set_category['category']['Uncategorized'][$key] = $item;
+        }
+        else {
+          $feature_set_category['category'][$item['#featuresetinfo']['category']][$key] = $item;
+        }
+      }
+      else {
+        $feature_set_category['form'][$key] = $item;
+      }
+    }
+
+    ksort($feature_set_category['category']);
+    $variables['feature_set_category'] = $feature_set_category;
+  }
+
+  $variables['feature_set_input'] = render($feature_set_category['form']['submit']) .
+    render($feature_set_category['form']['form_build_id']) .
+    render($feature_set_category['form']['form_token']) .
+    render($feature_set_category['form']['form_id']);
+
+  drupal_add_js('misc/form.js');
+  drupal_add_js('misc/collapse.js');
+}
