--- feature_set.admin.inc	2015-02-25 14:00:55.000000000 +0100
+++ feature_set.admin.inc	2015-07-01 15:23:39.000000000 +0200
@@ -33,6 +33,7 @@
             'featureset' => $details['title'],
             'description' => $details['description'],
             'icon' => empty($details['icon']) ? '' : $details['icon'],
+            'category' => empty($details['category']) ? '' : $details['category'],
           ),
         );
 
--- feature_set.module	2015-02-25 14:00:55.000000000 +0100
+++ feature_set.module	2017-04-18 13:47:20.000000000 +0200
@@ -268,6 +268,43 @@
     'feature_set_admin_form' => array(
       'file' => 'feature_set.admin.inc',
       'render element' => 'form',
+	  'template' => 'feature-set-admin-form',
     ),
   );
 }
+
+/**
+ * Implements template_preprocess().
+ */
+function feature_set_preprocess_feature_set_admin_form(&$variables) {
+  if (isset($variables['form']['#form_id']) && $variables['form']['#form_id'] == 'feature_set_admin_form') {
+    $feature_set_category = array();
+    foreach ($variables['form'] as $key => $item) {
+      $variables['feature_set_row'][$key] = $item;
+
+      // Get category if available.
+      if (is_array($item) && isset($item['#featuresetinfo'])) {
+        if (empty($item['#featuresetinfo']['category'])) {
+          $feature_set_category['category']['Uncategorized'][$key] = $item;
+        }
+        else {
+          $feature_set_category['category'][$item['#featuresetinfo']['category']][$key] = $item;
+        }
+      }
+      else {
+        $feature_set_category['form'][$key] = $item;
+      }
+    }
+
+    ksort($feature_set_category['category']);
+    $variables['feature_set_category'] = $feature_set_category;
+  }
+
+  $variables['feature_set_input'] = render($feature_set_category['form']['submit']) .
+    render($feature_set_category['form']['form_build_id']) .
+    render($feature_set_category['form']['form_token']) .
+    render($feature_set_category['form']['form_id']);
+
+  drupal_add_js('misc/form.js');
+  drupal_add_js('misc/collapse.js');
+}

--- feature-set-admin-form.tpl.php	1970-01-01 01:00:00.000000000 +0100
+++ feature-set-admin-form.tpl.php	2017-04-18 13:02:39.374697383 +0200
@@ -0,0 +1,50 @@
+<?php
+/**
+ * @file
+ * Theme implementation to display feature set.
+ *
+ * Available variables:
+ * - $feature_set_category: list of features, grouped by category
+ * - $feature_set_row: raw list of features, ungrouped
+ * - $feature_set_input: rendered form input (submit and hidden fields)
+ */
+?>
+<?php foreach ($feature_set_category['category'] as $category => $features) : ?>
+<fieldset class="collapsible form-wrapper collapsed">
+  <legend>
+    <span class="fieldset-legend"><?php print $category; ?></span>
+  </legend>
+  <div class="fieldset-wrapper">
+  <table>
+    <?php $i = 0; foreach ($features as $key => $item) : ?>
+      <tr class="<?php $i % 2 ? print 'even' : print 'odd';?>">
+        <td>
+          <?php
+            if (!empty($item['#featuresetinfo']['featureset'])) :
+              print '<strong>' . $item['#featuresetinfo']['featureset'] . '</strong>';
+            endif;
+            if (!empty($item['#featuresetinfo']['description'])) :
+              print '<br /><small>' . $item['#featuresetinfo']['description'] . '</small>';
+            endif;
+            if (!empty($item['#featuresetinfo']['description'])) :
+              print '<br /><small>' .
+                l(t('See @name documentation', array('@name' => $item['#featuresetinfo']['featureset'])),
+                  $item['#featuresetinfo']['documentation'],
+                  array('attributes' => array('target' => '_blank')))
+                . '</small>';
+            endif;
+          ?>
+        </td>
+        <td>
+          <?php print render($item); ?>
+        </td>
+      </tr>
+    <?php $i++; endforeach ?>
+  </table>
+  </div>
+</fieldset>
+<?php endforeach; ?>
+
+<?php
+  print $feature_set_input;
+?>
