--- FeedsXPathParserBase.inc
+++ FeedsXPathParserBase.inc
@@ -199,7 +199,7 @@ abstract class FeedsXPathParserBase extends FeedsParser {
     foreach ($mappings_ as $mapping) {
       if (strpos($mapping['source'], 'xpathparser:') === 0) {
         $mappings[$mapping['source']] = $mapping['target'];
-        if ($mapping['unique']) {
+        if (isset($mapping['unique']) && $mapping['unique']) {
           $uniques[] = $mapping['target'];
         }
       }
