diff --git a/modules/field/field.api.php b/modules/field/field.api.php
index b0fc6a4..52c6c2d 100644
--- modules/field/field.api.php
+++ modules/field/field.api.php
@@ -31,6 +31,8 @@
  *   - label: The human readable name of the element.
  *   - description: A short description of the element contents.
  *   - weight: The default weight of the element.
+ *   - visible: (optional, defaults to TRUE) The default visibility of the
+ *     display component. Not used for form components.
  *   - edit: (optional) String containing markup (normally a link) used as the
  *     element's 'edit' operation in the administration interface. Only for
  *     'form' context.
diff --git a/field.info.class.inc b/field.info.class.inc
index 3b89898..891fc8a 100644
--- modules/field/field.info.class.inc
+++ modules/field/field.info.class.inc
@@ -655,11 +655,12 @@ public function prepareExtraFields($extra_fields, $entity_type, $bundle) {
         else {
           $field_data['display'][$view_mode] = array(
             'weight' => $field_data['weight'],
-            'visible' => TRUE,
+            'visible' => isset($field_data['visible']) ? $field_data['visible'] : TRUE,
           );
         }
       }
       unset($field_data['weight']);
+      unset($field_data['visible']);
       $result['display'][$name] = $field_data;
     }
 
