diff --git a/flag.install b/flag.install
index 8451dce..d76f1e5 100644
--- a/flag.install
+++ b/flag.install
@@ -134,6 +134,12 @@ function flag_schema() {
   $schema['flag_types'] = array(
     'description' => 'The entity bundles that are affected by a flag.',
     'fields' => array(
+      'id' => array(
+        'description' => 'The unique identifier for this flag type.',
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ),
       'fid' => array(
         'description' => 'The unqiue flag ID as defined for the flag in {flag}.',
         'type' => 'int',
@@ -150,6 +156,7 @@ function flag_schema() {
         'default' => '',
       ),
     ),
+    'primary key' => array('id'),
     'indexes' => array(
       'fid' => array('fid'),
     ),
@@ -702,3 +709,23 @@ function flag_update_7306() {
     $flag->save();
   }
 }
+
+/**
+ * Add a primary key to flag_types table.
+ */
+function flag_update_7307() {
+  $db_table = 'flag_types';
+
+  if (!db_index_exists($db_table, 'PRIMARY')
+    && !db_field_exists($db_table, 'id')) {
+
+    db_add_field($db_table, 'id',
+      array(
+        'description' => 'Serial primary key.',
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ),
+      array('primary key' => array('id')));
+  }
+}
