diff --git a/flag.module b/flag.module
index 7b54c65..da35aef 100644
--- a/flag.module
+++ b/flag.module
@@ -527,15 +527,10 @@ function flag_hook_info() {
 function flag_fetch_definition($entity_type = NULL) {
   $definitions = &drupal_static(__FUNCTION__);
   if (!isset($definitions)) {
-    if ($cache = cache_get('flag_type_info')) {
-      $definitions = $cache->data;
-    }
-    else {
-      $definitions = module_invoke_all('flag_type_info');
-      drupal_alter('flag_type_info', $definitions);
-
-      cache_set('flag_type_info', $definitions);
-    }
+    module_load_include('inc', 'flag', 'flag.flag');
+    $definitions = module_invoke_all('flag_type_info');
+    drupal_alter('flag_type_info', $definitions);
+    cache_set('flag_type_info', $definitions);
   }
 
   if (isset($entity_type)) {
