diff --git a/honeypot.install b/honeypot.install
index e171681..1dacffa 100644
--- a/honeypot.install
+++ b/honeypot.install
@@ -12,11 +12,18 @@ function honeypot_schema() {
   $schema['honeypot_user'] = array(
     'description' => 'Table that stores failed attempts to submit a form.',
     'fields' => array(
+      'id'  => array(
+        'description' => 'Primary Key: row ID.',
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ),
       'uid' => array(
         'description' => 'Foreign key to {users}.uid; uniquely identifies a Drupal user to whom this ACL data applies.',
         'type' => 'int',
         'unsigned' => TRUE,
         'not null' => TRUE,
+        'default' => 0,
       ),
       'hostname' => array(
         'type' => 'varchar',
@@ -31,6 +38,7 @@ function honeypot_schema() {
         'not null' => TRUE,
       ),
     ),
+    'primary key' => array('id'),
     'indexes' => array(
       'uid' => array('uid'),
       'timestamp' => array('timestamp'),
@@ -184,3 +192,27 @@ function honeypot_update_7101() {
   // relying on the flood mechanism to track anonymous honeypot captures.
   flood_clear_event('honeypot');
 }
+
+/**
+ * Adds the 'id' column to the {honeypot_user} table and set primary key.
+ */
+function honeypot_update_7102() {
+  $db_table = 'honeypot_user';
+  $schema = honeypot_schema();
+
+  // Allow uid to set default 0.
+  db_change_field($db_table, 'uid', 'uid', $schema['honeypot_user']['fields']['uid']);
+
+  if (!db_index_exists($db_table, 'PRIMARY')
+    && !db_field_exists($db_table, 'id')) {
+
+    db_add_field($db_table, 'id',
+      array(
+        'description' => 'Serial primary key.',
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ),
+      array('primary key' => array('id')));
+  }
+}
