--- js/widget.js	2014-07-07 19:23:35.000000000 +0200
+++ js/widget.js	2015-07-28 10:13:47.281997339 +0200
@@ -7,6 +7,35 @@
       // Ensure we've set the default icon path to the leaflet library.
       L.Icon.Default.imagePath = settings.leaflet_widget.defaultIconPath;
 
+      $(document).ready(function() {
+        if (context == document) {
+          // If there are vertical tabs the widget should refresh when swapping between them.
+          if ($('.vertical-tabs').length > 0 && $('.vertical-tabs-panes').length > 0) {
+            var refresh = function() {
+              $('.vertical-tabs-panes').find('.vertical-tabs-pane').each(function(key, pane){
+                // Check pane is visible and refresh widget if it is.
+                if ($(pane).is(':visible')) {
+                  $(pane).find('.leaflet-widget').each(function(key, map) {
+                    var map_widget = Drupal.leaflet_widget[$(map).attr('id')];
+                    if(map_widget != undefined) {
+                      map_widget.invalidateSize();
+                      map_widget.fitBounds(map_widget.widget.vectors.getBounds());
+                    }
+                  });
+                }
+              });
+            };
+
+            // Refresh current vertical tab.
+            refresh();
+
+            // Refresh when changing to a different vertical tab.
+            $('.vertical-tabs').find('.vertical-tab-button').each(function(key, tab){
+              $(tab).find('a').bind('click', refresh);
+            });
+          }
+        }
+      });
+
       $('.leaflet-widget').once().each(function (i, item) {
         var id = $(item).attr('id'),
           options = settings.leaflet_widget_widget[id];
