--- includes/media_dailymotion.theme.inc	2014-06-18 12:41:41.370832147 +0200
+++ includes/media_dailymotion.theme.inc	2014-06-18 12:46:07.882334071 +0200
@@ -25,6 +25,13 @@
 
   $variables['wrapper_id'] = 'media_dailymotion_' . $variables['video_id'] . '_' . $variables['id'];
 
+  // handle protocol
+  global $is_https;
+  $protocol = 'http://';
+  if ($is_https) {
+    $protocol = 'https://';
+  }
+
   //TODO support background, foreground and highlight parameters?
   // For users with JavaScript, these object and embed tags will be replaced
   // by an iframe, so that we can support users without Flash.
@@ -34,13 +41,13 @@
       <param name="movie" value="http://www.dailymotion.com/swf/video/{$variables['video_id']}?autoPlay={$variables['autoplay']}"></param>
       <param name="allowFullScreen" value="{$variables['fullscreen']}"></param>
       <param name="allowScriptAccess" value="always"></param>
-      <embed type="application/x-shockwave-flash" src="http://www.dailymotion.com/swf/video/{$variables['video_id']}?autoPlay={$variables['autoplay']}" width="{$variables['width']}" height="{$variables['height']}" allowfullscreen="{$variables['fullscreen']}" allowscriptaccess="always"></embed>
+      <embed type="application/x-shockwave-flash" src="{$protocol}www.dailymotion.com/swf/video/{$variables['video_id']}?autoPlay={$variables['autoplay']}" width="{$variables['width']}" height="{$variables['height']}" allowfullscreen="{$variables['fullscreen']}" allowscriptaccess="always"></embed>
     </object>
 OUTPUT;
   }
   else {
     $variables['output'] = <<<OUTPUT
-    <iframe frameborder="0" width="{$variables['width']}" height="{$variables['height']}" src="http://www.dailymotion.com/embed/video/{$variables['video_id']}?logo=0&autoPlay={$variables['autoplay']}"></iframe>
+    <iframe frameborder="0" width="{$variables['width']}" height="{$variables['height']}" src="{$protocol}www.dailymotion.com/embed/video/{$variables['video_id']}?logo=0&autoPlay={$variables['autoplay']}"></iframe>
 OUTPUT;
   }
 }
