--- includes/media_flickr.utilities.inc.orig	2013-10-21 15:52:58.560619510 +0200
+++ includes/media_flickr.utilities.inc	2013-10-21 16:06:25.000000000 +0200
@@ -174,7 +174,7 @@
   if ($info && ($info['@attributes']['stat'] != 'fail')) {
     return $user_id;
   }
-  elseif ($info['err']['@attributes']['code'] == 1) {
+  elseif (isset($info['err']) && $info['err']['@attributes']['code'] == 1) {
     // See if we were passed a username instead.
     $info = _media_flickr_sets_request('flickr.people.findByUsername', array('username' => $user_id));
     if ($info && $info['@attributes']['stat'] != 'fail') {
