diff --git a/includes/media_avportal.formatters.inc b/includes/media_avportal.formatters.inc
index 33638f8..181b66d 100644
--- a/includes/media_avportal.formatters.inc
+++ b/includes/media_avportal.formatters.inc
@@ -53,9 +53,14 @@ function media_avportal_file_formatter_video_view($file, $display) {
     // Let's add the language to the options.
     global $language;
     $wrapper = file_stream_wrapper_get_instance_by_uri($file->uri);
-    $languages = $wrapper->getAvailableVideoLanguage();
+    try {
+      $languages = $wrapper->getAvailableVideoLanguage();
+    }
+    catch (Exception  $e) {
+      watchdog_exception('media_avportal', $e);
+    }
     $video_language = 'en';
-    if (in_array($language->language, $languages)) {
+    if (is_array($languages) && in_array($language->language, $languages)) {
       $video_language = $language->language;
     }
     $options['lg'] = $video_language;
@@ -113,7 +118,10 @@ function media_avportal_file_formatter_image_view($file, $display) {
   $wrapper = file_stream_wrapper_get_instance_by_uri($file->uri);
   $image_style = $display['settings']['image_style'];
   $valid_image_styles = image_style_options(FALSE);
-  $path = $wrapper->getLocalThumbnailPath();
+  $path = 'public://media-icons/default/video-x-generic.png';
+  if ($wrapper->getLocalThumbnailPath() != NULL) {
+    $path = $wrapper->getLocalThumbnailPath();
+  }
 
   if (empty($image_style) || !isset($valid_image_styles[$image_style])) {
     return array(
