diff --git a/includes/media_avportal.formatters.inc b/includes/media_avportal.formatters.inc
index 33638f8..277202f 100644
--- a/includes/media_avportal.formatters.inc
+++ b/includes/media_avportal.formatters.inc
@@ -53,7 +53,12 @@ function media_avportal_file_formatter_video_view($file, $display) {
     // Let's add the language to the options.
     global $language;
     $wrapper = file_stream_wrapper_get_instance_by_uri($file->uri);
-    $languages = $wrapper->getAvailableVideoLanguage();
+    try {
+      $languages = $wrapper->getAvailableVideoLanguage();
+    }
+    catch (Exception  $e) {
+      watchdog_exception('media_avportal', $e);
+    }
     $video_language = 'en';
     if (in_array($language->language, $languages)) {
       $video_language = $language->language;
@@ -113,7 +118,12 @@ function media_avportal_file_formatter_image_view($file, $display) {
   $wrapper = file_stream_wrapper_get_instance_by_uri($file->uri);
   $image_style = $display['settings']['image_style'];
   $valid_image_styles = image_style_options(FALSE);
-  $path = $wrapper->getLocalThumbnailPath();
+  try {
+      $path = $wrapper->getLocalThumbnailPath();
+  }
+  catch (Exception  $e) {
+    watchdog_exception('media_avportal', $e);
+  }
 
   if (empty($image_style) || !isset($valid_image_styles[$image_style])) {
     return array(
