diff --git a/modules/media_wysiwyg/media_wysiwyg.install b/modules/media_wysiwyg/media_wysiwyg.install
index 6a2f3ea..6fb972b 100644
--- a/modules/media_wysiwyg/media_wysiwyg.install
+++ b/modules/media_wysiwyg/media_wysiwyg.install
@@ -27,6 +27,7 @@ function media_wysiwyg_schema() {
         'default' => '',
       ),
     ),
+    'primary key' => array('type', 'display'),
   );
   $schema['media_view_mode_wysiwyg'] = array(
     'description' => 'Maps WYSIWYG view modes to file types.',
@@ -47,6 +48,7 @@ function media_wysiwyg_schema() {
       ),
 
     ),
+    'primary key' => array('type', 'view_mode'),
   );
   return $schema;
 }
@@ -254,3 +256,30 @@ function media_wysiwyg_update_7207() {
   $message = t('If you would like to be able to link images to a page, go to /admin/config/media/browser, check "Provide the ability to link media to pages", and save the settings.');
   drupal_set_message($message, 'warning', TRUE);
 }
+
+/**
+ * Add missing primary key.
+ */
+function media_wysiwyg_update_7208() {
+
+  $db_tables = array(
+    'media_restrict_wysiwyg' => array(
+      'type',
+      'display',
+    ),
+    'media_view_mode_wysiwyg' => array(
+      'type',
+      'view_mode',
+    ),
+  );
+
+  foreach ($db_tables as $db_table => $keys) {
+
+    // If not already done manually,
+    // set primary key.
+    if (!db_index_exists($db_table, 'PRIMARY')) {
+
+      db_add_primary_key($db_table, $keys);
+    }
+  }
+}
\ No newline at end of file
