--- includes/menu.inc.old	2013-11-07 15:58:49.212206348 +0100
+++ includes/menu.inc	2013-11-07 15:59:19.282343488 +0100
@@ -2591,6 +2591,7 @@
     }
 
     foreach ($active_trail as $parent) {
+      $parent['localized_options'] = (is_null($parent['localized_options']))?array():$parent['localized_options'];
       $breadcrumb[] = l($parent['title'], $parent['href'], $parent['localized_options']);
     }
   }
