diff --git a/mpdf.php b/mpdf.php
index aebeefb..caf1c84 100755
--- a/mpdf.php
+++ b/mpdf.php
@@ -90,11 +90,6 @@ $errorlevel = error_reporting($errorlevel & ~E_NOTICE);
 
 //error_reporting(E_ALL);
 
-if (function_exists("date_default_timezone_set")) {
-	if (ini_get("date.timezone") == "") {
-		date_default_timezone_set("Europe/London");
-	}
-}
 
 if (!function_exists('mb_strlen')) {
 	throw new MpdfException('mPDF requires mb_string functions. Ensure that mb_string extension is loaded.');
