--- simplenews.module	Mon Aug 04 14:11:25 2014
+++ simplenews.module	Mon Aug 04 14:13:39 2014
@@ -849,7 +849,7 @@
     $form['simplenews'] = array(
       '#type' => 'fieldset',
       '#description' => t('Select the newsletter(s) to which you wish to subscribe.'),
-      '#weight' => 5,
+      '#weight' => 98,
     );
     $form['simplenews']['newsletters'] = array(
       '#type' => 'checkboxes',
