diff --git a/redirect.module b/redirect.module
index 5d92399..311e7a7 100644
--- a/redirect.module
+++ b/redirect.module
@@ -1430,6 +1430,8 @@ function redirect_variables() {
 
 function redirect_parse_url($url) {
   $original_url = $url;
+  $is_external = url_is_external($url);
+  
   $url = trim($url, " \t\n\r\0\x0B\/");
   $parsed = parse_url($url);
 
@@ -1456,7 +1458,12 @@ function redirect_parse_url($url) {
     }
   }
 
-  $url = trim($url, '/');
+  // At this point, a URL with a query or fragment may still have a trailing
+  // slash. Trim the trailing slash from "internal" Drupal paths (and other
+  // local paths).
+  if (!$is_external) {
+    $url = trim($url, '/');
+  }
 
   // Convert to frontpage paths.
   if ($url == '<front>') {
