diff --git a/modules/taxonomy/taxonomy.install b/modules/taxonomy/taxonomy.install
index ecbd0f2e1a..a3825a1524 100644
--- a/modules/taxonomy/taxonomy.install
+++ b/modules/taxonomy/taxonomy.install
@@ -204,9 +204,9 @@ function taxonomy_schema() {
         'default'=> 0,
       ),
     ),
+    'primary key' => array('nid', 'tid'),
     'indexes' => array(
       'term_node' => array('tid', 'sticky', 'created'),
-      'nid' => array('nid'),
     ),
     'foreign keys' => array(
       'tracked_node' => array(
@@ -937,3 +937,41 @@ function taxonomy_update_7011(&$sandbox) {
 /**
  * @} End of "addtogroup updates-7.x-extra".
  */
+
+/**
+ * Add primary key and drop nid key.
+ */
+function taxonomy_update_7012() {
+  $db_table = 'taxonomy_index';
+
+  // Removal of the now superfluous index on nid.
+  if (db_index_exists($db_table, 'nid')) {
+    db_drop_index($db_table, 'nid');
+  }
+
+  // Create a temporary id
+  // allowing duplicate removal.
+  db_add_field($db_table, 'tempid', array(
+      'type' => 'serial',
+      'not null' => TRUE,
+      'description' => 'Tempid.',
+    ),
+    array(
+      'unique keys' => array(
+        'tempid' => array('tempid')
+      ),
+    )
+  );
+
+  // Remove duplicate.
+  db_query('DELETE t1 FROM {taxonomy_index} as t1 join {taxonomy_index} as t2 WHERE t1.nid = t2.nid AND t1.tid = t2.tid and t1.tempid > t2.tempid');
+  // Remove tempid.
+  db_drop_field($db_table, 'tempid');
+
+  // Now if not already done manually,
+  // set primary key.
+  if (!db_index_exists($db_table, 'PRIMARY')) {
+
+    db_add_primary_key($db_table, array('nid', 'tid'));
+  }
+}
diff --git a/modules/taxonomy/taxonomy.module b/modules/taxonomy/taxonomy.module
index e3ee48e0cf..4e3e32fc40 100644
--- a/modules/taxonomy/taxonomy.module
+++ b/modules/taxonomy/taxonomy.module
@@ -1973,16 +1973,12 @@ function taxonomy_build_node_index($node) {
     }
     // Insert index entries for all the node's terms.
     if (!empty($tid_all)) {
-      $query = db_insert('taxonomy_index')->fields(array('nid', 'tid', 'sticky', 'created'));
       foreach ($tid_all as $tid) {
-        $query->values(array(
-          'nid' => $node->nid,
-          'tid' => $tid,
-          'sticky' => $sticky,
-          'created' => $node->created,
-        ));
+        db_merge('taxonomy_index')
+          ->key(array('nid' => $node->nid, 'tid' => $tid))
+          ->fields(array('sticky' => $sticky, 'created' => $node->created))
+          ->execute();
       }
-      $query->execute();
     }
   }
 }
