diff --git a/token.module b/token.module
index 56e3ceb..175d408 100644
--- a/token.module
+++ b/token.module
@@ -838,6 +838,13 @@ function token_build_tree($token_type, array $options = array()) {
     $token_type = $entity_token_type;
   }
 
+  // Avoid massive memory consumption.
+  // http://drupal.org/node/1058912
+  // http://drupal.org/node/1005364
+  if ($token_type == 'user' || $token_type == 'node') {
+    $options['depth'] = min($options['depth'], 2);
+  }
+
   $tree_cid = "tree:{$token_type}:{$language->language}:{$options['depth']}";
 
   // If we do not have this base tree in the static cache, check {cache_token}
