--- modules/user/user.pages.inc	2013-12-02 11:14:09.548837109 +0100
+++ modules/user/user.pages.inc	2013-12-02 15:21:26.000000000 +0100
@@ -68,7 +68,9 @@
     form_set_value(array('#parents' => array('account')), $account, $form_state);
   }
   else {
-    form_set_error('name', t('Sorry, %name is not recognized as a user name or an e-mail address.', array('%name' => $name)));
+    drupal_set_message(t('Further instructions have been sent to your e-mail address.'));
+    form_set_error('form_build_id', t(''));
+drupal_goto('user');
   }
 }
 
