--- handlers/views_handler_filter.inc
+++ handlers/views_handler_filter.inc
@@ -602,15 +602,19 @@ class views_handler_filter extends views_handler {
     }

     if (!empty($form_state['values']['options']['group_info']['group_items'])) {
+      $operators = $this->operators();
       foreach ($form_state['values']['options']['group_info']['group_items'] as $id => $group) {
         if (empty($group['remove'])) {

           // Check if the title is defined but value wasn't defined.
           if (!empty($group['title'])) {
-            if ((!is_array($group['value']) && trim($group['value']) == "") ||
-                (is_array($group['value']) && count(array_filter($group['value'], '_views_array_filter_zero')) == 0)) {
-              form_error($form['group_info']['group_items'][$id]['value'],
-                         t('The value is required if title for this item is defined.'));
+            // Check if the operator requires a value
+            if ($operators[$group['operator']]['values'] > 0) {
+              if ((!is_array($group['value']) && trim($group['value']) == "") ||
+                  (is_array($group['value']) && count(array_filter($group['value'], '_views_array_filter_zero')) == 0)) {
+                form_error($form['group_info']['group_items'][$id]['value'],
+                           t('The value is required if title for this item is defined.'));
+              }
             }
           }

