diff --git a/includes/ajax.inc b/includes/ajax.inc
index bef3c05..b7ba03a 100644
--- a/includes/ajax.inc
+++ b/includes/ajax.inc
@@ -40,7 +40,7 @@ function views_ajax() {
 
     // Load the view.
     $view = views_get_view($name);
-    if ($view && $view->access($display_id)) {
+    if ($view && $view->access($display_id) && $view->set_display($display_id) && $view->display_handler->use_ajax()) {
       // Fix 'q' for paging.
       if (!empty($path)) {
         $_GET['q'] = $path;
