diff --git a/js/ajax_view.js b/js/ajax_view.js
index 7d54c67..4eb8b34 100644
--- a/js/ajax_view.js
+++ b/js/ajax_view.js
@@ -56,8 +56,9 @@ Drupal.views.ajaxView = function(settings) {
 
   this.settings = settings;
 
-  // Add the ajax to exposed forms.
-  this.$exposed_form = this.$view.children('.view-filters').children('form');
+  // Add the ajax to exposed forms. We look into the view filters element as a
+  // fallback in case the default markup was altered.
+  this.$exposed_form = this.$view.find('form.views-exposed-form, .view-filters form');
   this.$exposed_form.once(jQuery.proxy(this.attachExposedFormAjax, this));
 
   // Add the ajax to pagers.
diff --git a/views.module b/views.module
index 9cdcfb5..a35845a 100644
--- a/views.module
+++ b/views.module
@@ -2031,7 +2031,7 @@ function views_exposed_form($form, &$form_state) {
   $form['#action'] = url($view->display_handler->get_url());
   $form['#theme'] = views_theme_functions('views_exposed_form', $view, $display);
   $form['#id'] = drupal_clean_css_identifier('views_exposed_form-' . check_plain($view->name) . '-' . check_plain($display->id));
-//  $form['#attributes']['class'] = array('views-exposed-form');
+  $form['#attributes']['class'] = array('views-exposed-form');
 
   // If using AJAX, we need the form plugin.
   if ($view->use_ajax) {
