diff --git a/workbench_og.module b/workbench_og.module
index dc2f6ca..d82a5f8 100755
--- a/workbench_og.module
+++ b/workbench_og.module
@@ -117,6 +117,23 @@ function workbench_og_node_access_records($node) {
       );
     }
   }
+  else {
+    // Assign grants to unpublished nodes that belong to no group.
+    $roles = _workbench_og_view_authorized_roles();
+    foreach ($roles as $key => $value) {
+      $grants[] = array(
+        // Realm to view unpublished content.
+        'realm' => 'workbench_moderation_og_view_all_unpublished',
+        // Assign role gid.
+        'gid' => $key,
+        'grant_view' => 1,
+        'grant_update' => 0,
+        'grant_delete' => 0,
+        'priority' => 2,
+      );
+    }
+  }
+
   return $grants;
 }
 
@@ -149,7 +166,16 @@ function workbench_og_node_grants($account, $op) {
   if (user_access('view all unpublished content')) {
     // Assign grant to access content withe the user uid as gid.
     $grants['workbench_moderation_og_view_own_unpublished'][] = $account->uid;
+    // Assign grant to access unpublished content.
+    $roles = _workbench_og_view_authorized_roles();
+    $enabled_roles = array_intersect($roles, $account->roles);
+    if (count($enabled_roles) > 0) {
+      foreach ($enabled_roles as $key => $value) {
+        $grants['workbench_moderation_og_view_all_unpublished'][] = $key;
+      }
+    }
   }
+
   return $grants;
 }
 
@@ -170,3 +196,23 @@ function workbench_og_help($path, $arg) {
       }
   }
 }
+
+/**
+ * Get roles authorized to see unpublished content.
+ *
+ * @return array
+ *   Array with roles ['rid' => 'role name']
+ */
+function _workbench_og_view_authorized_roles() {
+  $defined_roles = user_roles();
+  $authorized_roles = array();
+
+  foreach ($defined_roles as $key => $value) {
+    $roles = user_role_permissions(array($key => $value));
+    if (isset($roles[$key]['view all unpublished content']) && $roles[$key]['view all unpublished content'] == 1) {
+      $authorized_roles[$key] = $value;
+    }
+  }
+
+  return $authorized_roles;
+}
